<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>MOFA model on aIg stimulated BJAB cells</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manuscript QuRIE-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="QC.html">QC</a>
</li>
<li>
  <a href="MOFAaIg.html">MOFA aIg</a>
</li>
<li>
  <a href="MOFAibru.html">MOFA ibru</a>
</li>
<li>
  <a href="license.html">License and citation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MOFA model on aIg stimulated BJAB cells</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>QuRIE-seq_manuscript/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201117code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201117)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201117code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201117)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomvanbuggenumQuRIEseqmanuscripttree4f4828dc0b105e22d522002a43d0f6b17d11bacdtargetblank4f4828da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/tree/4f4828dc0b105e22d522002a43d0f6b17d11bacd" target="_blank">4f4828d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomvanbuggenumQuRIEseqmanuscripttree4f4828dc0b105e22d522002a43d0f6b17d11bacdtargetblank4f4828da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/tree/4f4828dc0b105e22d522002a43d0f6b17d11bacd" target="_blank">4f4828d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/raw/
    Ignored:    output/MOFA_aIg.hdf5
    Ignored:    output/MOFA_ibru.hdf5
    Ignored:    output/data_weights_prot_fact1and3.csv
    Ignored:    output/metadata_Ricard.csv

Unstaged changes:
    Modified:   analysis/revisions.rmd
    Modified:   output/seu_aIG_samples.rds
    Modified:   output/seu_combined_filtered_normalized.rds
    Modified:   output/seu_ibru_samples.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/MOFAaIg.Rmd</code>) and HTML (<code>docs/MOFAaIg.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/4f4828dc0b105e22d522002a43d0f6b17d11bacd/analysis/MOFAaIg.Rmd" target="_blank">4f4828d</a>
</td>
<td>
jessievb
</td>
<td>
2021-07-08
</td>
<td>
update header and small edits html output
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/c51a9e773353e491548a8cd9e2985235d0b8cce8/docs/MOFAaIg.html" target="_blank">c51a9e7</a>
</td>
<td>
jessievb
</td>
<td>
2021-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/3a318781bd95f41b96e98757f46c7cb7e9feeefa/docs/MOFAaIg.html" target="_blank">3a31878</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2021-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/f77904258d5f38666176a90994bd8f8765f7f91e/analysis/MOFAaIg.Rmd" target="_blank">f779042</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2021-06-02
</td>
<td>
update suppl fig 8 pSYK overlay
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/8e7e92aa7548d8ec70be22b98f47489c4bde3da3/docs/MOFAaIg.html" target="_blank">8e7e92a</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2021-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/5c8adc2261eb560805dc036c106d93921d2b1e2c/analysis/MOFAaIg.Rmd" target="_blank">5c8adc2</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2021-06-02
</td>
<td>
update suppl fig WNN
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/bec19cbe4eb74f44c8b177c470f532687b067a00/docs/MOFAaIg.html" target="_blank">bec19cb</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2021-06-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/0f790122aacb60e0fa2e54b240213a59ee964fb5/analysis/MOFAaIg.Rmd" target="_blank">0f79012</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2021-06-01
</td>
<td>
Added WNN and pJAK1 high vs low supplementary figures revisions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/372d628d77e4b8a979369e2092792b87bc3f2156/docs/MOFAaIg.html" target="_blank">372d628</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/0ac5ddb20dc35dd454ab2017ca02d5c96bc6366a/analysis/MOFAaIg.Rmd" target="_blank">0ac5ddb</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-12-01
</td>
<td>
typo panel title
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/07b4d2089419938ac48f29866b6776a555f547c1/docs/MOFAaIg.html" target="_blank">07b4d20</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/f6b2e5d25419be3fd893e3e088ac924a2cec69ba/analysis/MOFAaIg.Rmd" target="_blank">f6b2e5d</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-12-01
</td>
<td>
small edits panel titles
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/d8ba888f410990268a92e0b3549a7990a82037db/docs/MOFAaIg.html" target="_blank">d8ba888</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/1d2b544254b3c41a666d5f2ffb29ce4f28601809/analysis/MOFAaIg.Rmd" target="_blank">1d2b544</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-26
</td>
<td>
minor change to capitalization of protein names
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/bbd89400dffad818f87f12acfc4988908c993636/docs/MOFAaIg.html" target="_blank">bbd8940</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/be4b6104c29a28a40c7814c605dc3b86945baba7/analysis/MOFAaIg.Rmd" target="_blank">be4b610</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-26
</td>
<td>
Capitalize several proteins in all plots (hardcoded)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/61cb95458bab3b29188e8b72d1e020232a085299/docs/MOFAaIg.html" target="_blank">61cb954</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/cf646e039da3fd7c93cff01ff549f47338b6d4c9/analysis/MOFAaIg.Rmd" target="_blank">cf646e0</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-25
</td>
<td>
typo Syk to SYK
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/47ecaed94474bfd9d71d03049c448dd3457480c8/docs/MOFAaIg.html" target="_blank">47ecaed</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/14c0ebf2831e464136125855b1d78cfbf533fffe/analysis/MOFAaIg.Rmd" target="_blank">14c0ebf</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
small update documentation github page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/416c6aeeb52f4ba7775eda7e0681c572d30344be/docs/MOFAaIg.html" target="_blank">416c6ae</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/2290bafdd6c02e1188d9ece4467c326195f0db8a/analysis/MOFAaIg.Rmd" target="_blank">2290baf</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
hide figure panels in html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/f95d1d2fd13c485d64a93f74f1882e51dfc4fa87/docs/MOFAaIg.html" target="_blank">f95d1d2</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/6b16185a88693c643e53e69d161a8810eeddafc0/analysis/MOFAaIg.Rmd" target="_blank">6b16185</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
update minor changes to suppl fig mofa
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/df6d45f219c5d06b2f1be5b9ce106c75d6534df1/docs/MOFAaIg.html" target="_blank">df6d45f</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/a96d5ba425ae8f6bded33f2e72d8825346c51595/analysis/MOFAaIg.Rmd" target="_blank">a96d5ba</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
update minor changes to documentation aIg
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/38267b11dbec4d2fed8dde09ac0f3ba7cf23f2e6/docs/MOFAaIg.html" target="_blank">38267b1</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/8b81996c8ed528b0a0ae85d5ad0e9836a6d225bd/analysis/MOFAaIg.Rmd" target="_blank">8b81996</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
update minor changes to documentation &amp; figure1/2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/aed3aab4af8ee447b9cfa9b2957d74344112d973/docs/MOFAaIg.html" target="_blank">aed3aab</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/b1037ae30772e7a6de5d30e7fafc147e72c4ac1a/analysis/MOFAaIg.Rmd" target="_blank">b1037ae</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-23
</td>
<td>
update suppl fig MOFA aIG
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/8c8e40e25d19a8bbc0c2b21fde1f2cfd728ffe54/docs/MOFAaIg.html" target="_blank">8c8e40e</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/b8950ba20a62c9e6e012893d62c26e26a6943d6f/analysis/MOFAaIg.Rmd" target="_blank">b8950ba</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-20
</td>
<td>
update mofa plot axis orientation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/081882dfa0292fdeede3da3721bca3680bc43762/docs/MOFAaIg.html" target="_blank">081882d</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/0cda3dc17e090caebd65040a9e601f7631e1d07e/analysis/MOFAaIg.Rmd" target="_blank">0cda3dc</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-19
</td>
<td>
Figure 1 and 2 plus supplementary figures manuscript
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/2ffe38d8ad99572f5b5e0fdde94cbd8db50659bf/docs/MOFAaIg.html" target="_blank">2ffe38d</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/4487750fd014ae3af0bfaa3cd6d581cf1ac3392c/analysis/MOFAaIg.Rmd" target="_blank">4487750</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-17
</td>
<td>
basis set-up of page display
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/d9372d117842a51e92f9ba0293d6e92c8e9b29c1/analysis/MOFAaIg.Rmd" target="_blank">d9372d1</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-17
</td>
<td>
Publish the initial files for this website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(
  message = F, warning = F, echo = T, eval = T
)

source(&quot;code/load_packages.R&quot;)
seu_combined_selectsamples &lt;- readRDS(&quot;output/seu_aIG_samples.rds&quot;)
library(ggrepel)

panellabels &lt;- c(&#39;a&#39;, &#39;b&#39;, &#39;c&#39;,&#39;d&#39; , &#39;e&#39;, &#39;f&#39;, &#39;g&#39;, &#39;h&#39;, &#39;i&#39;, &#39;j&#39;, &#39;k&#39;)

add.textsize &lt;- theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 7),
          axis.text.y = element_text(colour = &#39;black&#39;,size=7),
          text = element_text(size=7),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7)
          )


colorgradient6_manual &lt;- c(&quot;#F7FBFF&quot;,&quot;#CFE1F2&quot;, &quot;#93C4DE&quot;, &quot;#4A97C9&quot;, &quot;#1F5284&quot;, &quot;#0C2236&quot; )
colorgradient6_manual2 &lt;- c(&quot;#d4d4d3&quot;,&quot;#CFE1F2&quot;, &quot;#93C4DE&quot;, &quot;#4A97C9&quot;, &quot;#1F5284&quot;, &quot;#0C2236&quot; )
labels &lt;- c(&quot;0&quot;, &quot;2&quot;, &quot;4&quot;, &quot;6&quot;, &quot;60&quot;, &quot;180&quot;)</code></pre>
<div id="mofa-analysis-time-series-aig" class="section level2">
<h2>MOFA analysis time-series aIg</h2>
<div id="variable-features" class="section level4">
<h4>Variable features</h4>
<p>Determine variable genes and proteins</p>
<pre class="r"><code>seu_combined_selectsamples &lt;- FindVariableFeatures(seu_combined_selectsamples, selection.method = &quot;mvp&quot;, assay = &quot;SCT.RNA&quot;, num.bin =20, mean.cutoff = c(0, 5), dispersion.cutoff = c(0.5,10), nfeatures =500)

genes.variable &lt;- seu_combined_selectsamples@assays$SCT.RNA@var.features[-grep(&quot;^MT&quot;, seu_combined_selectsamples@assays$SCT.RNA@var.features)] # without the mitochondrial genes 

proteins.all &lt;- row.names(seu_combined_selectsamples[[&quot;PROT&quot;]])

meta.allcells &lt;- seu_combined_selectsamples@meta.data %&gt;%
  mutate(sample = rownames(seu_combined_selectsamples@meta.data))</code></pre>
</div>
<div id="mofa-model" class="section level4">
<h4>MOFA model</h4>
<p>Note, if mofa object was generated before, it will read the generated rds file. (this will speed-up the process of generating this html document if edits are required)</p>
<pre class="r"><code>## Note, if mofa object was generated before, it will read the generated rds file. (this will speed-up the process of generating this html document if edits are required)

myfiles &lt;- list.files(path=&quot;output/&quot;, pattern = &quot;.rds$&quot;)

if(&quot;MOFA_aIg.rds&quot; %in%  myfiles){mofa &lt;- readRDS(&quot;output/MOFA_aIg.rds&quot;)} else { #If so, read object, else do:
     
      mofa &lt;- create_mofa(list(
        &quot;RNA&quot; = as.matrix( seu_combined_selectsamples@assays$SCT.RNA@scale.data[genes.variable,] ),
        &quot;PROT&quot; = as.matrix( seu_combined_selectsamples@assays$PROT@scale.data[proteins.all,] ))
        )

      # Default settings used (try 15 factors, excludes all non-informative factors)
      data_opts &lt;- get_default_data_options(mofa)
      model_opts &lt;- get_default_model_options(mofa)
      train_opts &lt;- get_default_training_options(mofa)
      train_opts$seed &lt;- 42 # use same seed for reproducibility
      mofa &lt;- prepare_mofa(
        object = mofa,
        data_options = data_opts,
        model_options = model_opts,
        training_options = train_opts
        )

    mofa &lt;- run_mofa(mofa, outfile = &quot;output/MOFA_aIg.hdf5&quot;)
    mofa &lt;- run_umap(mofa, factors = c(1:7))
    samples_metadata(mofa) &lt;- meta.allcells
    saveRDS(mofa, file= &quot;output/MOFA_aIg.rds&quot;)
  
}


mofa</code></pre>
<pre><code>Trained MOFA with the following characteristics: 
 Number of views: 2 
 Views names: RNA PROT 
 Number of features (per view): 2159 80 
 Number of groups: 1 
 Groups names: group1 
 Number of samples (per group): 4754 
 Number of factors: 7 </code></pre>
<pre class="r"><code>## Hack to rename protein names for visualization
featurenamesmofa &lt;- features_names(mofa)

## todo cleanup/more efficient
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-Src&quot; ,&quot;p-SRC&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;Syk&quot; ,&quot;SYK&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-c-Jun&quot; ,&quot;p-cJUN&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;Akt&quot; ,&quot;AKT&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;Btk&quot; ,&quot;BTK&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-Akt&quot; ,&quot;p-AKT&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-Syk&quot; ,&quot;p-SYK&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;Erk1/2&quot; ,&quot;ERK1/2&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-Erk1/2&quot; ,&quot;p-ERK1/2&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-Myc&quot; ,&quot;p-MYC&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;p-Btk&quot; ,&quot;p-BTK&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;Bcl-6&quot; ,&quot;BCL6&quot;,x), how = &quot;replace&quot;)

featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;CD53-PROT&quot; ,&quot;CD53&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;CD70-PROT&quot; ,&quot;CD70&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;KLF6-PROT&quot; ,&quot;KLF6&quot;,x), how = &quot;replace&quot;)
featurenamesmofa &lt;- rapply(featurenamesmofa,function(x) ifelse(x==&quot;XBP1-PROT&quot; ,&quot;XBP1&quot;,x), how = &quot;replace&quot;)


features_names(mofa) &lt;- featurenamesmofa</code></pre>
<pre class="r"><code># Export Factor 1 and 3 weights for Cytoscape plot
data_weights_prot &lt;- get_weights(object = mofa, views = &quot;PROT&quot;, factors = c(1,3), as.data.frame = TRUE)

data_weights_prot &lt;- as.data.frame(data_weights_prot)  %&gt;%
  spread(factor, value) %&gt;%
  mutate(Factor1_signflip = Factor1 *-1)

write.csv2(data_weights_prot, file = &quot;output/data_weights_prot_fact1and3.csv&quot;)</code></pre>
</div>
</div>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<div id="umap-on-7-mofa-factors" class="section level3">
<h3>UMAP on 7 MOFA factors</h3>
<pre class="r"><code>## UMAP
plot.umap.data &lt;-  plot_dimred(mofa, method=&quot;UMAP&quot;, color_by = &quot;condition&quot;,stroke = 0.001, dot_size =1, alpha = 0.2, return_data = T)

plot.umap.all &lt;- ggplot(plot.umap.data, aes(x=x, y = y, fill = color_by))+
  geom_point(size = 0.8, alpha = 0.6, shape = 21, stroke = 0) +
  theme_half_open() +
  scale_fill_manual(values = colorgradient6_manual, labels = c(labels), name = &quot;Time aIg&quot;,)+
   theme(legend.position=&quot;none&quot;)+
  scale_x_reverse()+
  scale_y_reverse()+
  add.textsize +
  labs(title = &quot;UMAP on MOFA+ factors shows minute and \nhour timescale signal transductions&quot;, x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;)

## UMAP legend
legend.umap &lt;- get_legend( ggplot(plot.umap.data, aes(x=x, y = y, fill = color_by))+
  geom_point(size = 2, alpha = 1, shape = 21, stroke = 0) +
  theme_half_open() +
  scale_fill_manual(values = colorgradient6_manual, labels = c(labels), name = &quot;Time aIg \n(minutes)&quot;,)+
  add.textsize +
  labs(title = &quot;UMAP on MOFA+ factors shows minute and \nhour timescale signal transductions&quot;, x = &quot;UMAP 1&quot;, &quot;y = UMAP 2&quot;))
legend.umap &lt;- as_ggplot(legend.umap)</code></pre>
<pre class="r"><code>plot_fig1_umap &lt;- plot_grid(plot.umap.all,legend.umap, labels = c(&quot;&quot;), label_size = 10, ncol = 2, rel_widths = c(1, 0.2))

ggsave(plot_fig1_umap, filename = &quot;output/paper_figures/Fig1_UMAP.pdf&quot;, width = 68, height = 62, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(plot_fig1_umap, filename = &quot;output/paper_figures/Fig1_UMAP.png&quot;, width = 68, height = 62, units = &quot;mm&quot;,  dpi = 300)

plot_fig1_umap</code></pre>
<p><img src="figure/MOFAaIg.Rmd/Fig1_UMAP-1.png" width="288" style="display: block; margin: auto;" /> <em>Figure 1.</em> UMAP of 7 MOFA factors, integrating phospho-protein and RNA measurements</p>
</div>
<div id="suppl-pca-wnn" class="section level3">
<h3>Suppl PCA &amp; WNN</h3>
<pre class="r"><code>seu_combined_selectsamples &lt;- RunPCA(seu_combined_selectsamples,assay = &quot;SCT.RNA&quot;, features = genes.variable, verbose = FALSE, ndims.print = 0, reduction.name = &quot;pca.RNA&quot;)
seu_combined_selectsamples &lt;- RunPCA(seu_combined_selectsamples, assay = &quot;PROT&quot;, features = proteins.all, verbose = FALSE, ndims.print = 0,reduction.name = &quot;pca.PROT&quot;)</code></pre>
<pre class="r"><code>## PCA analysis

plot.PCA.RNA &lt;- DimPlot(seu_combined_selectsamples, reduction = &quot;pca.RNA&quot;, group.by = &quot;condition&quot;, pt.size =0.2) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg&quot;,)+
  labs(title = &quot;RNA PCA separates cells \nat hour timescale&quot;, x= &quot;RNA PC 1&quot;, y = &quot; RNA PC 2&quot;) +
  add.textsize +
  theme(legend.position = &quot;none&quot;)


plot.PCA.PROT &lt;- DimPlot(seu_combined_selectsamples, reduction = &quot;pca.PROT&quot;, group.by = &quot;condition&quot;, pt.size = 0.2) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg&quot;,)+
  labs(title = &quot;(Phospho)protein PCA separates cells\nat minutes timescale&quot;, x= &quot;Protein PC 1&quot;, y = &quot;Protein PC 2&quot;) +
  add.textsize +
  #  scale_x_reverse()+
  theme(legend.position = &quot;none&quot;)

legend &lt;- get_legend(DimPlot(seu_combined_selectsamples, reduction = &quot;pca.RNA&quot;, group.by = &quot;condition&quot;, pt.size =0.1) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg&quot;,)+
  add.textsize)
legend &lt;- as_ggplot(legend)


PCA.PROTPC1.data &lt;- data.frame(rank = 1:80, 
                               protein = names(sort(seu_combined_selectsamples@reductions$pca.PROT@feature.loadings[,1])),
                               weight.PC1 = sort(seu_combined_selectsamples@reductions$pca.PROT@feature.loadings[,1]),
                               highlights = c(names(sort(seu_combined_selectsamples@reductions$pca.PROT@feature.loadings[,1]))[1:4],rep(&quot;&quot;,76))
                               )

PCA.PROTPC1.data &lt;- PCA.PROTPC1.data %&gt;%
  mutate(highlights = as.character(highlights)) %&gt;%
  mutate(highlights = case_when(as.character(highlights) == &quot;p-Syk&quot; ~ &quot;p-SYK&quot;,
                                as.character(highlights) == &quot;p-Src&quot; ~ &quot;p-SRC&quot;,
                           TRUE ~ highlights))

plot.PCA.PROTweightsPC1 &lt;- ggplot(PCA.PROTPC1.data, aes(x=weight.PC1, y = rank, label = highlights)) +
  geom_point(size=0.1) +
  labs(title = &quot;Top 4 protein loadings \n&quot;, x= &quot;Weight Protein PC1&quot;) +
  geom_point(color = ifelse(PCA.PROTPC1.data$highlights == &quot;&quot;, &quot;grey50&quot;, &quot;red&quot;)) +
  geom_text_repel(size = 1.5, segment.size = 0.25)+
  theme_half_open()+
  add.textsize +
  theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
        ) +
  scale_x_reverse()+
  scale_y_reverse()


PCA.RNAPC1.data &lt;- data.frame(rank = 1:length(genes.variable), 
                               RNA = names(sort(seu_combined_selectsamples@reductions$pca.RNA@feature.loadings[,1])),
                               weight.PC1 = sort(seu_combined_selectsamples@reductions$pca.RNA@feature.loadings[,1]),
                               highlights = c(rep(&quot;&quot;,(length(genes.variable)-4)),names(sort(seu_combined_selectsamples@reductions$pca.RNA@feature.loadings[,1]))[(length(genes.variable)-3):length(genes.variable)])
                               ) 
## Loadings RNA
plot.PCA.RNAweightsPC1 &lt;- ggplot(PCA.RNAPC1.data, aes(x=weight.PC1, y = rank, label = highlights)) +
  geom_point(size=0.1) +
  labs(title = &quot;Top 4 RNA loadings \n&quot;, x= &quot;Weight RNA PC1&quot;) +
  geom_point(color = ifelse(PCA.RNAPC1.data$highlights == &quot;&quot;, &quot;grey50&quot;, &quot;red&quot;)) +
  geom_text_repel(size = 1.5, segment.size = 0.25)+
  theme_half_open()+
  add.textsize +
  theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
        ) 

## ridgeplots
PC1.data &lt;- data.frame(sample = rownames(seu_combined_selectsamples@reductions$pca.PROT@cell.embeddings),
                       PC1_PROT = seu_combined_selectsamples@reductions$pca.PROT@cell.embeddings[,1],
                       PC1_RNA = seu_combined_selectsamples@reductions$pca.RNA@cell.embeddings[,1]) %&gt;%
  left_join(meta.allcells)

plot_ridge_PC1Prot &lt;- ggplot(PC1.data, aes(x = PC1_PROT, y = condition, fill = condition)) + 
  scale_fill_manual(values = colorgradient6_manual, labels = c(labels), name = &quot;Time aIg \n(minutes)&quot;)+
  geom_density_ridges2() +
  scale_y_discrete(labels = labels,  name = &quot;Time aIg (minutes)&quot;)+
  scale_x_continuous(name = &quot;PC1 Proteins&quot;) +
  theme_half_open() +
  add.textsize+
  theme(legend.position = &quot;none&quot;)

plot_ridge_PC1RNA &lt;- ggplot(PC1.data, aes(x = PC1_RNA, y = condition, fill = condition)) + 
  scale_fill_manual(values = colorgradient6_manual, labels = c(labels), name = &quot;Time aIg \n(minutes)&quot;)+
  geom_density_ridges2() +
  scale_y_discrete(labels = labels,  name = &quot;Time aIg (minutes)&quot;)+
  scale_x_continuous(name = &quot;PC1 RNA&quot;) +
  theme_half_open() +
  add.textsize+
  theme(legend.position = &quot;none&quot;)</code></pre>
<pre class="r"><code>seu_combined_selectsamples &lt;- FindMultiModalNeighbors(
  seu_combined_selectsamples, reduction.list = list(&quot;pca.RNA&quot;, &quot;pca.PROT&quot;), 
  dims.list = list(c(1:7), c(1:7)), modality.weight.name = &quot;RNA.weight&quot;
)


seu_combined_selectsamples &lt;- RunUMAP(seu_combined_selectsamples, nn.name = &quot;weighted.nn&quot;, reduction.name = &quot;wnn.umap&quot;, reduction.key = &quot;wnnUMAP_&quot;)

WNN.Dimplot &lt;- DimPlot(seu_combined_selectsamples, reduction = &#39;wnn.umap&#39;, label = FALSE, repel = TRUE, label.size = 2.5) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg&quot;,)+
  labs(title = &quot;WNN UMAP similar to \n MOFA+ based UMAP&quot;, x= &quot;wnnUMAP 1&quot;, y = &quot;wnnUMAP 2&quot;) +
  add.textsize +
  #  scale_x_reverse()+
  theme(legend.position = &quot;right&quot;)</code></pre>
<pre class="r"><code>Factorvalues &lt;- data.frame(get_factors(mofa)[1])[,1:4]
colnames(Factorvalues) &lt;- c(paste0(&quot;Factor&quot;, 1:4))
seu_combined_selectsamples &lt;- AddMetaData(seu_combined_selectsamples, Factorvalues)

plot.wnn.Factor1  &lt;- FeaturePlot(seu_combined_selectsamples, reduction = &quot;wnn.umap&quot;, features = &quot;Factor1&quot;) +
  scale_color_gradient2(name = &quot;Factor 1 \nvalue&quot;) +
  labs(title = &quot;MOFA+ Factor 1 &amp; 3 overlap \nwith WNN UMAP&quot;, x= &quot;wnnUMAP 1&quot;, y = &quot;wnnUMAP 2&quot;) +
  add.textsize +
  theme(legend.position = &quot;right&quot;)

plot.wnn.Factor3  &lt;- FeaturePlot(seu_combined_selectsamples, reduction = &quot;wnn.umap&quot;, features = &quot;Factor3&quot;) +
  scale_color_gradient2(name = &quot;Factor 3 \nvalue&quot;) +
  labs(title = &quot;&quot;, x= &quot;wnnUMAP 1&quot;, y = &quot;wnnUMAP 2&quot;) +
  add.textsize +
  #  scale_x_reverse()+
  theme(legend.position = &quot;right&quot;)

plot.wnn.Factor1  &lt;- FeaturePlot(seu_combined_selectsamples, reduction = &quot;wnn.umap&quot;, features = &quot;Factor1&quot;) +
  scale_color_gradient2(name = &quot;Factor 1 \nvalue&quot;) +
  labs(title = &quot;MOFA+ Factor 1 &amp; 3 overlap \nwith WNN UMAP&quot;, x= &quot;wnnUMAP 1&quot;, y = &quot;wnnUMAP 2&quot;) +
  add.textsize +
  theme(legend.position = &quot;right&quot;)

plot.wnn.pSYK  &lt;- FeaturePlot(seu_combined_selectsamples, reduction = &quot;wnn.umap&quot;, features = &quot;p-Syk&quot;, slot = &quot;scale.data&quot;) +
  scale_color_gradient2(name = &quot;p-SYK \nscaled counts&quot;) +
  labs(title = &quot;\n&quot;, x= &quot;wnnUMAP 1&quot;, y = &quot;wnnUMAP 2&quot;) +
  add.textsize +
  #  scale_x_reverse()+
  theme(legend.position = &quot;right&quot;)</code></pre>
<pre class="r"><code>Fig1.pca &lt;- plot_grid(plot.PCA.PROT,plot.PCA.RNA,legend, plot_ridge_PC1Prot, plot_ridge_PC1RNA,NULL,  plot.PCA.PROTweightsPC1, plot.PCA.RNAweightsPC1,NULL,WNN.Dimplot,plot.wnn.pSYK,NULL,plot.wnn.Factor1,plot.wnn.Factor3, labels = c(panellabels[1:2], &quot;&quot;,panellabels[3:4],&quot;&quot;,panellabels[5:6],&quot;&quot;,panellabels[7:8],&quot;&quot;,panellabels[9:10]), label_size = 10, ncol = 3, rel_widths = c(0.9,0.9,0.2,0.9,0.9), rel_heights = c(1,0.8,1.3,1))

ggsave(filename = &quot;output/paper_figures/Suppl_PCA_aIg_wnn.pdf&quot;, width = 143, height = 226, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(filename = &quot;output/paper_figures/Suppl_PCA_aIg_wnn.png&quot;, width = 143, height = 226, units = &quot;mm&quot;,  dpi = 300)

Fig1.pca</code></pre>
<p><img src="figure/MOFAaIg.Rmd/Figure_1_supplementary-1.png" width="537.6" style="display: block; margin: auto;" /> <em>Supplementary Figure.</em> PCA analysis on RNA and Protein datasets separately.</p>
</div>
<div id="suppl-mofa-model-properties" class="section level3">
<h3>Suppl MOFA model properties</h3>
<pre class="r"><code>## Variance per factor
plot.variance.perfactor.all &lt;- plot_variance_explained(mofa, x=&quot;factor&quot;, y=&quot;view&quot;) +
    add.textsize +
    labs(title = &quot;Variance explained by each factor per modality&quot;) 

## variance total
plot.variance.total &lt;- plot_variance_explained(mofa, x=&quot;view&quot;, y=&quot;factor&quot;, plot_total = T)
plot.variance.total &lt;- plot.variance.total[[2]] +
  add.textsize +
    labs(title = &quot;Total \nvariance&quot;) +
  geom_text(aes(label=round(R2,1)), vjust=1.6, color=&quot;white&quot;, size=2.5)

## Significance correlation covariates
plot.heatmap.pval.covariates &lt;- as.ggplot(correlate_factors_with_covariates(mofa, 
  covariates = c(&quot;time&quot;),
  factors = 1:mofa@dimensions$K,
  fontsize = 7, 
  cluster_row = F,
  cluster_col = F
))+ 
  add.textsize +
  theme(axis.text.y=element_blank(),
        axis.text.x=element_blank(),
        plot.title = element_text(size=7, face = &quot;plain&quot;),
        )</code></pre>
<pre class="r"><code>## Factor values over time
plot.violin.factorall &lt;- plot_factor(mofa, 
  factor = c(2,4:mofa@dimensions$K),
  color_by = &quot;condition&quot;,
  dot_size = 0.2,      # change dot size
  dodge = T,           # dodge points with different colors
  legend = F,          # remove legend
  add_violin = T,      # add violin plots,
  violin_alpha = 0.9  # transparency of violin plots
) +   
  add.textsize +
  scale_color_manual(values=c(colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg&quot;)) +
  scale_fill_manual(values=c(colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg&quot;))+
  labs(title = &quot;Factor values per time-point of additional factors not correlating with time.&quot; ) +
  theme(axis.text.x=element_blank())

## Loadings factors stable protein

plot.rank.PROT.2.4to7 &lt;- plot_weights(mofa, 
  view = &quot;PROT&quot;, 
  factors = c(1:mofa@dimensions$K), 
  nfeatures = 3, 
  text_size = 1.5
) +   
  add.textsize +
  labs(title = &quot;Top 3 Protein loadings per factor&quot;) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
)

## Loadings factors stable protein

plot.rank.RNA.2.4to7 &lt;- plot_weights(mofa, 
  view = &quot;RNA&quot;, 
  factors = c(1:mofa@dimensions$K), 
  nfeatures = 3, 
  text_size = 1.5
) +   
  add.textsize +
  labs(title = &quot;Top 3 RNA loadings per factor&quot;) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
)

## correlation time and factors
plot.correlation.covariates &lt;- correlate_factors_with_covariates(mofa, 
  covariates = c(&quot;time&quot;),
  factors = mofa@dimensions$K:1,
  plot = &quot;r&quot;
)</code></pre>
<pre class="r"><code>plot.correlation.covariates &lt;- ggcorrplot(plot.correlation.covariates, tl.col = &quot;black&quot;, method = &quot;square&quot;, lab = TRUE, ggtheme = theme_void, colors = c(&quot;#11304C&quot;, &quot;white&quot;, &quot;#11304C&quot;), lab_size = 2.5) +
  add.textsize +
  labs(title = &quot;Correlation of \nfactors with \ntime of treatment&quot;, y = &quot;&quot;) +
  scale_y_discrete(labels = &quot;&quot;) +
  coord_flip() +
  theme(axis.text.x=element_text(angle =0,hjust = 0.5), 
        axis.text.y=element_text(size = 5),
        legend.position=&quot;none&quot;,
        plot.title = element_text(hjust = 0.5))</code></pre>
<pre class="r"><code>Fig.1.suppl.mofa.row1 &lt;- plot_grid(plot.variance.perfactor.all, plot.variance.total,NULL, plot.heatmap.pval.covariates, labels = c(panellabels[1:3]), label_size = 10, ncol = 4, rel_widths = c(1.35, 0.3,0.25,0.38))

Fig.1.suppl.mofa.row2 &lt;- plot_grid(plot.violin.factorall,legend.umap, labels = panellabels[4], label_size = 10, ncol = 2, rel_widths = c(1,0.1))

Suppl_mofa &lt;- plot_grid(Fig.1.suppl.mofa.row1, Fig.1.suppl.mofa.row2, plot.rank.PROT.2.4to7, plot.rank.RNA.2.4to7, labels = c(&quot;&quot;,&quot;&quot;, panellabels[5:6]),label_size = 10, ncol = 1, rel_heights = c(1.45,1,1.1,1.1))


ggsave(Suppl_mofa, filename = &quot;output/paper_figures/Fig2.Suppl_MOFAaIg.pdf&quot;, width = 183, height = 220, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(Suppl_mofa, filename = &quot;output/paper_figures/Fig2.Suppl_MOFAaIg.png&quot;, width = 183, height = 220, units = &quot;mm&quot;,  dpi = 300)

Suppl_mofa</code></pre>
<p><img src="figure/MOFAaIg.Rmd/Suppl_MOFA_model-1.png" width="691.2" style="display: block; margin: auto;" /> <em>Supplementary Figure.</em> MOFA model additional information</p>
</div>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<p>Prepare main panels</p>
<pre class="r"><code>meta.allcells &lt;- seu_combined_selectsamples@meta.data %&gt;%
  mutate(sample = rownames(seu_combined_selectsamples@meta.data))

proteindata &lt;- as.data.frame(t(seu_combined_selectsamples@assays$PROT@scale.data)) %&gt;%
  mutate(sample = rownames(t(seu_combined_selectsamples@assays$PROT@scale.data))) %&gt;%
  left_join(meta.allcells)

MOFAfactors&lt;- as.data.frame(mofa@expectations$Z) %&gt;%
  mutate(sample = rownames(as.data.frame(mofa@expectations$Z)[,1:mofa@dimensions$K])) 

MOFAfactors &lt;- left_join(as.data.frame(MOFAfactors), meta.allcells)

weights.prot &lt;- get_weights(mofa, views = &quot;PROT&quot;,as.data.frame = TRUE)

topnegprots.factor1 &lt;- weights.prot %&gt;%
  filter(factor == &quot;Factor1&quot; &amp; value &lt;= 0) %&gt;%
  arrange(value)

topposprots.factor3 &lt;- weights.prot %&gt;%
  filter(factor == &quot;Factor3&quot; &amp; value &gt;= 0) %&gt;%
  arrange(-value)


weights.RNA &lt;- get_weights(mofa, views = &quot;RNA&quot;,as.data.frame = TRUE)</code></pre>
<pre class="r"><code>## violin plots phospho-proteins highlighted in main

f.violin.fact &lt;- function(data , protein){
  
  ggplot(subset(data)  , aes(x = as.factor(condition), y =get(noquote(protein)))) +
    annotate(&quot;rect&quot;,
          xmin = 5 - 0.45,
             xmax = 6 + 0.6,
           ymin = -4.5, ymax = 4.5, fill = &quot;lightblue&quot;,
           alpha = .4
  )+
    geom_hline(yintercept=0, linetype=&#39;dotted&#39;, col = &#39;black&#39;)+ 
  geom_violin(alpha = 0.9,aes(fill = condition))+
  geom_jitter(width = 0.05,size = 0.1, fill = &quot;black&quot;, shape = 21)+ 
  stat_summary(fun=median, geom=&quot;point&quot;, shape=95, size=2, inherit.aes = T, position = position_dodge(width = 0.9), color = &quot;red&quot;)+
  theme_few()+
  ylab(paste0(protein)) +
  scale_x_discrete(labels = labels, expand = c(0.1,0.1), name = &quot;Time aIg (minutes)&quot;) +
    scale_y_continuous(expand = c(0,0), name = strsplit(protein, split = &quot;\\.&quot;)[[1]][2]) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg &quot;,)+ 
  scale_fill_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg &quot;,) +
  add.textsize +
  theme(#axis.title.x=element_blank(),
        #axis.text.x=element_blank(),
        #axis.ticks.x=element_blank(),
        legend.position=&quot;none&quot;) 
}

factors_toplot &lt;- c(colnames(MOFAfactors)[c(1,3)])

for(i in factors_toplot) {
assign(paste0(&quot;plot.factor.&quot;, i), f.violin.fact(data = MOFAfactors,protein = i)) 
}

legend.violinfactor &lt;- as_ggplot( get_legend( ggplot(subset(MOFAfactors)  , aes(x = as.factor(condition), y =get(noquote(factors_toplot[1])))) +
    annotate(&quot;rect&quot;,
          xmin = 5 - 0.45,
             xmax = 6 + 0.6,
           ymin = -4.5, ymax = 4.5, fill = &quot;lightblue&quot;,
           alpha = .4
  )+
    geom_hline(yintercept=0, linetype=&#39;dotted&#39;, col = &#39;black&#39;)+ 
  geom_violin(alpha = 0.9,aes(fill = condition))+
  geom_jitter(width = 0.05,size = 0.1, aes(col = condition), fill = &quot;black&quot;, shape = 21)+ 
  stat_summary(fun=median, geom=&quot;point&quot;, shape=23, size=2, inherit.aes = T, position = position_dodge(width = 0.9), color = &quot;black&quot;)+
  theme_few()+
  ylab(paste0(factors_toplot[1])) +
  scale_x_discrete(labels = labels, expand = c(0.1,0.1), name = &quot;Time aIg  (minutes)&quot;) +
    scale_y_continuous(expand = c(0,0), name = strsplit(factors_toplot[1], split = &quot;\\.&quot;)[[1]][2]) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg \n(minutes)&quot;,)+ 
  scale_fill_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg \n(minutes)&quot;,) +
  add.textsize ) ) 

plot.violin.factor1 &lt;- plot.factor.group1.Factor1 +
  scale_y_reverse(expand = c(0,0), name = &quot;Factor 1 value&quot;)+
  annotate(geom=&quot;text&quot;, x=1.5, y=-4.3, label=&quot;minutes&quot;, size = 2,
              color=&quot;grey3&quot;) +
  annotate(geom=&quot;text&quot;, x=5.5, y=-4.3, label=&quot;hours&quot;, size = 2,
              color=&quot;grey3&quot;) +
  labs(title = &quot;Factor 1 captures \nminute timescale signal transduction&quot;)

plot.violin.factor3 &lt;- plot.factor.group1.Factor3 +
  annotate(geom=&quot;text&quot;, x=5.5, y=4.3, label=&quot;hours&quot;, size = 2,
              color=&quot;grey3&quot;)+
  annotate(geom=&quot;text&quot;, x=1.5, y=4.3, label=&quot;minutes&quot;, size = 2,
              color=&quot;grey3&quot;) +
  scale_y_continuous(expand = c(0,0), name = &quot;Factor 3 value&quot;)+
  labs(title = &quot;Factor 3 captures \nhour timescale signal transduction&quot;)

#### protein Loadings factor 1

list.bcellpathway.protein &lt;- c(&quot;p-CD79a&quot;, &quot;p-SYK&quot;, &quot;p-SRC&quot;, &quot;p-ERK1/2&quot;, &quot;p-PLC-y2&quot;, &quot;p-BLNK&quot;,&quot;p-PLC-y2Y759&quot;,&quot;p-PKC-b1&quot;, &quot;p-p38&quot;, &quot;p-AKT&quot;, &quot;p-S6&quot;, &quot;p-TOR&quot;, &quot;CD79a&quot;) # , &quot;p-PLC-y2&quot;, &quot;p-PKC-b1&quot;, &quot;p-IKKa/b&quot;,&quot;p-JNK&quot;, &quot;p-p38&quot;, &quot;p-p65&quot;,&quot;p-Akt&quot;, &quot;p-S6&quot;, &quot;p-TOR&quot;
list.top20 &lt;- topnegprots.factor1$feature[1:20]
## protein factor 1
plotdata.rank.PROT.1 &lt;-plot_weights(mofa, 
  view = &quot;PROT&quot;, 
  factors = c(1), 
  nfeatures = 15, 
  text_size = 1,
  manual = list(list.top20, list.bcellpathway.protein, &quot;p-JAK1&quot; ), 
  color_manual = list(&quot;grey&quot;,&quot;blue4&quot;,&quot;red&quot;),
  return_data = TRUE
)

plotdata.rank.PROT.1 &lt;- plotdata.rank.PROT.1 %&gt;%
  mutate(Rank = 1:80,
         Weight = value, 
         colorvalue = ifelse(labelling_group == 2 &amp; value &lt;= -0.2,&quot;grey&quot;, ifelse(labelling_group == 3 &amp; value &lt;= -0.2, &quot;blue4&quot;, ifelse(labelling_group == 4 &amp; value &lt;= -0.2, &quot;red&quot;, &quot;grey&quot;))),
         highlights = ifelse(labelling_group &gt;= 1 &amp; value &lt;= -0.25, as.character(feature), &quot;&quot;)
         ) 

plot.rank.PROT.1 &lt;- ggplot(plotdata.rank.PROT.1, aes(x=Weight, y = Rank, label = highlights)) +
  labs(title = &quot;(Phospho)proteins: &lt;p&gt;&lt;span style=&#39;color:blue4&#39;&gt;BCR &lt;/span&gt;&amp;&lt;span style=&#39;color:red&#39;&gt; JAK1&lt;/span&gt; signaling&quot;, #&lt;span style=&#39;color:blue4&#39;&gt;BCR signaling&lt;/span&gt; and &lt;span style=&#39;color:red&#39;&gt;p-JAK1&lt;/span&gt;
       x= &quot;Factor 1 loading value&quot;,
       y= &quot;Factor 1 loading rank&quot;) +
  geom_point(size=0.1, color =plotdata.rank.PROT.1$colorvalue, size =1) +
  geom_text_repel(size = 2, 
                  segment.size = 0.2, 
                  color =plotdata.rank.PROT.1$colorvalue,
                  nudge_x       = 1 - plotdata.rank.PROT.1$Weight,
                  direction     = &quot;y&quot;,
                  hjust         = 1,
                  segment.color = &quot;grey50&quot;)+
  theme_few()+
  add.textsize  +
  scale_y_continuous(trans = &quot;reverse&quot;) +
  add.textsize +
  theme(
    plot.title = element_markdown()
  ) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
        )+
  xlim(c(-1,1))


## violin plots phospho-proteins highlighted in main

f.violin.prot &lt;- function(data , protein){
  
  ggplot(subset(data)  , aes(x = as.factor(condition), y =get(noquote(protein)))) +
    annotate(&quot;rect&quot;,
          xmin = 5 - 0.45,
             xmax = 6 + 0.6,
           ymin = -5.5, ymax = 5.5, fill = &quot;lightblue&quot;,
           alpha = .4
  )+
  geom_violin(alpha = 0.9,aes(fill = condition))+
  geom_jitter(width = 0.05,size = 0.1, color = &quot;black&quot;)+ 
  stat_summary(fun=median, geom=&quot;point&quot;, shape=95, size=2, inherit.aes = T, position = position_dodge(width = 0.9), color = &quot;red&quot;)+
  theme_few()+
  ylab(paste0(protein)) +
  scale_x_discrete(labels = labels, expand = c(0.1,0.1), name = &quot;Time aIg  (minutes)&quot;) +
  scale_y_continuous(expand = c(0,0), name = protein) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg &quot;,)+ 
  scale_fill_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg &quot;,) +
  add.textsize +
  theme(#axis.title.x=element_blank(),
        #axis.text.x=element_blank(),
        #axis.ticks.x=element_blank(),
        legend.position=&quot;none&quot;) 
}

proteins_toplot &lt;- c(&quot;p-CD79a&quot;,&quot;p-Syk&quot;,&quot;p-JAK1&quot;, &quot;IgM&quot;)

for(i in proteins_toplot) {
assign(paste0(&quot;plot.violin.&quot;, i), f.violin.prot(data = proteindata ,protein = i)) 

}

## Fig 2 bottom row panels
`plot.violin.p-CD79a` &lt;- `plot.violin.p-CD79a` +
  labs(title = &quot;BCR signaling pathway and \nJAK1 activation&quot;)+
  add.textsize+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),)

`plot.violin.p-Syk` &lt;- `plot.violin.p-Syk` +
  add.textsize+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),) +
  scale_y_continuous(name = &quot;p-SYK&quot;)

`plot.violin.p-JAK1` &lt;-`plot.violin.p-JAK1` +
  add.textsize +
  labs(y = &quot;&lt;span style=&#39;color:red&#39;&gt;p-JAK1&lt;/span&gt;&quot;) +
  theme(axis.title.y = element_markdown()) </code></pre>
<p>Enrichment analysis of Factor 3 positive loadings</p>
<pre class="r"><code>topposrna.factor3 &lt;- weights.RNA %&gt;%
  filter(factor == &quot;Factor3&quot; &amp; value &gt;= 0) %&gt;%
  arrange(-value)
rownames(topposrna.factor3) &lt;- topposrna.factor3$feature

### Convert Gene-names to gene IDs (using &#39;org.Hs.eg.db&#39; library)

topposrna.factor3 &lt;- topposrna.factor3 %&gt;%
  mutate(ENTREZID = mapIds(org.Hs.eg.db, as.character(topposrna.factor3$feature), &#39;ENTREZID&#39;, &#39;SYMBOL&#39;))

listgenes.factor3 &lt;- topposrna.factor3$value

names(listgenes.factor3) &lt;- topposrna.factor3$ENTREZID


go.pb.fct3.pos &lt;- enrichGO(gene         = topposrna.factor3$feature,
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;SYMBOL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;)

go.pb.fct3.pos &lt;- simplify(go.pb.fct3.pos, cutoff=0.6, by=&quot;p.adjust&quot;, select_fun=min)


go.pb.fct3.pos.dplyr &lt;- mutate(go.pb.fct3.pos, richFactor = Count / as.numeric(sub(&quot;/\\d+&quot;, &quot;&quot;, BgRatio))) 

## plot main panel
plot.enriched.go.bp.factor3 &lt;- ggplot(go.pb.fct3.pos.dplyr, showCategory = 10, 
  aes(-log10(p.adjust), fct_reorder(Description,  -log10(p.adjust)))) +   geom_segment(aes(xend=0, yend = Description)) +
  geom_point(aes(size = Count)) +
  scale_color_viridis_c(guide=guide_colorbar(reverse=TRUE)) +
  scale_size_continuous(range=c(0.5, 3)) +
  theme_minimal() + 
  add.textsize +
  xlab(&quot;-log adj pval&quot;) +
  ylab(NULL) + 
  ggtitle(&quot;Enriched Biological Processes&quot;) +
  theme(legend.position=&quot;none&quot;)

legend.plot.enriched.go.bp.factor3 &lt;- as.ggplot(get_legend(ggplot(go.pb.fct3.pos.dplyr, showCategory = 10, 
  aes(-log10(p.adjust), fct_reorder(Description,  -log10(p.adjust)))) +   geom_segment(aes(xend=0, yend = Description)) +
  geom_point(aes(size = Count)) +
  scale_color_viridis_c(guide=guide_colorbar(reverse=TRUE)) +
  scale_size_continuous(range=c(0.5, 3)) +
  theme_minimal() + 
  add.textsize +
  xlab(&quot;-log adj pval&quot;) +
  ylab(NULL) + 
  ggtitle(&quot;Enriched Biological Processes&quot;) +
     theme(legend.position=&quot;right&quot;)))

## Plot supplementary
plot.enriched.go.bp.factor3_50 &lt;- ggplot(go.pb.fct3.pos.dplyr, showCategory = 50, 
  aes(-log10(p.adjust), fct_reorder(Description,  -log10(p.adjust)))) +   geom_segment(aes(xend=0, yend = Description)) +
  geom_point(aes(size = Count)) +
  scale_color_viridis_c(guide=guide_colorbar(reverse=TRUE)) +
  scale_size_continuous(range=c(0.5, 3)) +
  theme_minimal() + 
  add.textsize +
  xlab(&quot;-log adj pval&quot;) +
  ylab(NULL) + 
  ggtitle(&quot;Enriched Biological Processes&quot;) </code></pre>
<pre class="r"><code>message(c(&quot;Significance protein folding GO term: \n&quot;,go.pb.fct3.pos.dplyr@result[which(go.pb.fct3.pos.dplyr@result$Description == &quot;protein folding&quot;),&quot;p.adjust&quot;]))</code></pre>
<pre class="r"><code>##For RNA loadings panels
topgeneset.fct3&lt;- unlist(str_split(go.pb.fct3.pos.dplyr@result[1:10,&quot;geneID&quot;], pattern = &quot;/&quot;))

topgeneset.fct3 = bitr(topgeneset.fct3, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Hs.eg.db&quot;)

## RNA factor 1 loadings
plotdata.rank.RNA.3 &lt;-plot_weights(mofa, 
  view = &quot;RNA&quot;, 
  factors = c(3), 
  nfeatures = 5, 
  text_size = 1,
  manual = list(topposrna.factor3$feature[c(1:5)] ,topgeneset.fct3$SYMBOL), 
  color_manual = list(&quot;grey&quot;,&quot;blue4&quot;),
  return_data = TRUE
)

plotdata.rank.RNA.3 &lt;- plotdata.rank.RNA.3 %&gt;%
  mutate(Rank = 1:nrow(plotdata.rank.RNA.3),
         Weight = value, 
         colorvalue = ifelse(labelling_group == 3 &amp;value &gt;= 0.25,&quot;blue4&quot;, ifelse(labelling_group == 2&amp;value &gt;= 0.25, &quot;grey2&quot;, &quot;grey&quot;)),
         highlights = ifelse(labelling_group &gt;= 1&amp;value &gt;= 0.25, as.character(feature), &quot;&quot;)
         )

plot.rank.RNA.3 &lt;- ggplot(plotdata.rank.RNA.3, aes(x=Weight, y = Rank, label = highlights)) +
  labs(title = &quot;Contributing genes &lt;p&gt;&lt;span style=&#39;color:blue4&#39;&gt;&lt;span style=&#39;color:blue4&#39;&gt;GO-term vesicle related &lt;/span&gt; &quot;,  #&lt;span style=&#39;color:blue4&#39;&gt;BCR signaling regulation (find GO term+list todo)&lt;/span&gt;
       x= &quot;Factor 3 loading value&quot;,
       y= &quot;Factor 3 loading rank&quot;) +
  geom_point(size=0.1, color =plotdata.rank.RNA.3$colorvalue) +
  geom_text_repel(size = 2, 
                  segment.size = 0.2, 
                  color =plotdata.rank.RNA.3$colorvalue,
                  nudge_x       = -1 - plotdata.rank.RNA.3$Weight,
                  direction     = &quot;y&quot;,
                  hjust         = 0,
                  segment.color = &quot;grey50&quot;)+
  theme_few()+
  add.textsize +
  scale_y_continuous() +
  add.textsize +
  theme(
    plot.title = element_markdown()
  )  +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.title.y = element_blank())+
  xlim(c(-1,1))


#### Protein loadings factor 3

list.highlight.prot.fct3 &lt;- c(&quot;p-p38&quot;, &quot;p-AKT&quot;, &quot;p-S6&quot;, &quot;p-TOR&quot;, &quot;XBP1_PROT&quot;, &quot;p-STAT5&quot;)
list.top20 &lt;- topnegprots.factor1$feature[1:20]

plotdata.rank.PROT.3 &lt;-plot_weights(mofa, 
  view = &quot;PROT&quot;, 
  factors = c(3), 
  nfeatures = 30, 
  text_size = 1,
  manual = list(topposprots.factor3$feature[c(1:8)] ,list.highlight.prot.fct3), 
  color_manual = list(&quot;grey&quot;,&quot;blue4&quot;),
  return_data = TRUE
)

plotdata.rank.PROT.3 &lt;- plotdata.rank.PROT.3 %&gt;%
  mutate(Rank = 1:80,
         Weight = value, 
         colorvalue = ifelse(labelling_group == 3 &amp;value &gt;= 0.25,&quot;blue4&quot;, ifelse(labelling_group == 2&amp;value &gt;= 0.4, &quot;grey&quot;, &quot;grey&quot;)),
         highlights = ifelse(labelling_group &gt;= 1&amp;value &gt;= 0.25, as.character(feature), &quot;&quot;)
         )  %&gt;%
  mutate(highlights = case_when(as.character(highlights) == &quot;XBP1_PROT&quot; ~ &quot;XBP1&quot;,
                           TRUE ~ highlights))

plot.rank.PROT.3 &lt;- ggplot(plotdata.rank.PROT.3, aes(x=Weight, y = Rank, label = highlights)) +
  labs(title = &quot;Signaling implicated in &lt;p&gt;&lt;span style=&#39;color:blue4&#39;&gt;Unfolded protein response&lt;/span&gt;&quot;, 
       x= &quot;Factor 3 loading value&quot;,
       y= &quot;Factor 3 loading rank&quot;) +
  geom_point(size=0.1, color =plotdata.rank.PROT.3$colorvalue, size =1) +
  geom_text_repel(size = 2, 
                  segment.size = 0.2, 
                  color =plotdata.rank.PROT.3$colorvalue,
                  nudge_x       = -1 - plotdata.rank.PROT.3$Weight,
                  direction     = &quot;y&quot;,
                  hjust         = 0,
                  segment.color = &quot;grey50&quot;)+
  theme_few()+
  add.textsize  +
  scale_y_continuous() +
  add.textsize +
  theme(
    plot.title = element_markdown()
  ) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
        )+
  xlim(c(-1,1))


#### RNA loadings factor 1

plotdata.rank.RNA.1 &lt;-plot_weights(mofa, 
  view = &quot;RNA&quot;, 
  factors = c(1), 
  nfeatures = 2, 
  text_size = 1,
  return_data = TRUE
)

plotdata.rank.RNA.1 &lt;- plotdata.rank.RNA.1 %&gt;%
  mutate(Rank = 1:nrow(plotdata.rank.RNA.1),
         Weight = value, 
         colorvalue = ifelse(labelling_group == 1,&quot;blue4&quot;, ifelse(labelling_group == 1, &quot;blue4&quot;, &quot;grey&quot;)),
         highlights = ifelse(labelling_group &gt;= 1, as.character(feature), &quot;&quot;)
         )

plot.rank.RNA.1 &lt;- ggplot(plotdata.rank.RNA.1, aes(x=Weight, y = Rank, label = highlights)) +
  labs(title = &quot; Contributing genes&lt;p&gt;&lt;span style=&#39;color:blue4&#39;&gt;BCR activation&lt;/span&gt;&quot;,  
       x= &quot;Factor 1 loading value&quot;,
       y= &quot;Factor 1 loading rank&quot;) +
  geom_point(size=0.1, color =plotdata.rank.RNA.1$colorvalue) +
  geom_text_repel(size = 2, 
                  segment.size = 0.2, 
                  color =plotdata.rank.RNA.1$colorvalue,
                  nudge_x       = 1 - plotdata.rank.RNA.1$Weight,
                  direction     = &quot;y&quot;,
                  hjust         = 1,
                  segment.color = &quot;grey50&quot;) +
  theme_few()+
  add.textsize +
  scale_y_continuous(trans = &quot;reverse&quot;) +
  add.textsize +
  theme(
    plot.title = element_markdown()
  )  +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank(), 
        axis.title.y=element_blank()
        )+
  xlim(c(-1,1))


#### Violins genes


f.violin.rna &lt;- function(data , protein){
  
  ggplot(subset(data)  , aes(x = as.factor(condition), y =get(noquote(protein)))) +
    annotate(&quot;rect&quot;,
          xmin = 5 - 0.45,
             xmax = 6 + 0.6,
           ymin = -3, ymax = 15, fill = &quot;lightblue&quot;,
           alpha = .4
  )+
  geom_violin(alpha = 0.9,aes(fill = condition))+
  geom_jitter(width = 0.05,size = 0.1, color = &quot;black&quot;)+ 
  stat_summary(fun=median, geom=&quot;point&quot;, shape=95, size=2, inherit.aes = T, position = position_dodge(width = 0.9), color = &quot;red&quot;)+
  theme_few()+
  ylab(paste0(protein)) +
  scale_x_discrete(labels = labels, expand = c(0.1,0.1), name = &quot;Time aIg  (minutes)&quot;) +
  scale_y_continuous(expand = c(0,0), name = protein) +
  scale_color_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg &quot;,)+ 
  scale_fill_manual(values = colorgradient6_manual2, labels = c(labels), name = &quot;Time aIg &quot;,) +
  add.textsize +
  theme(#axis.title.x=element_blank(),
        #axis.text.x=element_blank(),
        #axis.ticks.x=element_blank(),
        legend.position=&quot;none&quot;) 
}
rnadata &lt;- as.data.frame(t(seu_combined_selectsamples@assays$SCT.RNA@scale.data)) %&gt;%
  mutate(sample = rownames(t(seu_combined_selectsamples@assays$SCT.RNA@scale.data))) %&gt;%
  left_join(meta.allcells)

enriched.geneset.posregBcell.fact1 &lt;- c(&quot;NPM1&quot;, &quot;NEAT1&quot;, &quot;BTF3&quot;, &quot;IGHM&quot;, &quot;IGKC&quot;)

##Violin genes
for(i in enriched.geneset.posregBcell.fact1) {
assign(paste0(&quot;plot.violin.rna.&quot;, i), f.violin.rna(data = rnadata ,protein = i)) 
}

plot.violin.rna.NEAT1 &lt;- plot.violin.rna.NEAT1 +
  labs(title = &quot;Expression of upregulated genes\n&quot;)+
  add.textsize+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),)

plot.violin.rna.NPM1 &lt;- plot.violin.rna.NPM1 +
  add.textsize+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),)</code></pre>
<div id="main-figure-2" class="section level3">
<h3>Main figure 2</h3>
<pre class="r"><code>Fig2.row1.violinsprot &lt;- plot_grid(`plot.violin.p-CD79a`, `plot.violin.p-Syk`, `plot.violin.p-JAK1`,labels = c(panellabels[4]), label_size = 10, ncol = 1, rel_heights = c(1.2,0.95,1.2))
Fig2.row1 &lt;- plot_grid(plot.violin.factor1,plot.rank.PROT.1, NULL, Fig2.row1.violinsprot, labels = c(panellabels[1:3], &quot;&quot;),  label_size = 10, ncol =4, rel_widths = c(0.8,0.55,0.5,0.8))

Fig2.row2.violinsrna &lt;- plot_grid(plot.violin.rna.NEAT1,plot.violin.rna.NPM1, plot.violin.rna.BTF3,labels = &quot;&quot;, label_size = 10, ncol = 1, rel_heights = c(1.2,0.95,1.2))

Fig2.row2 &lt;- plot_grid(plot.violin.factor3,plot.rank.PROT.3,plot.rank.RNA.3,Fig2.row2.violinsrna, labels = c(panellabels[5:6], &quot;&quot;, panellabels[8]), label_size = 10, ncol =4, rel_widths = c(0.8,0.55,0.5,0.8))

Fig2.row3 &lt;- plot_grid(plot.enriched.go.bp.factor3,legend.plot.enriched.go.bp.factor3,NULL, labels = panellabels[7], label_size = 10, ncol =3, rel_widths = c(1.8,0.2,0.6))

plot_fig2 &lt;- plot_grid(Fig2.row1, Fig2.row2,Fig2.row3, labels = c(&quot;&quot;, &quot;&quot;, &quot;&quot;), label_size = 10, ncol = 1, rel_heights = c(1,1,0.55))

ggsave(plot_fig2, filename = &quot;output/paper_figures/Fig2.pdf&quot;, width = 183, height = 183, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(plot_fig2, filename = &quot;output/paper_figures/Fig2.png&quot;, width = 183, height = 183, units = &quot;mm&quot;,  dpi = 300)

plot_fig2</code></pre>
<p><img src="figure/MOFAaIg.Rmd/plot_fig2-1.png" width="691.2" style="display: block; margin: auto;" /> <em>Figure 2.</em> Factor 1 and 3 exploration.</p>
</div>
<div id="suppl-enriched-fact-3" class="section level3">
<h3>Suppl Enriched Fact 3</h3>
<pre class="r"><code>ggsave(plot.enriched.go.bp.factor3_50, filename = &quot;output/paper_figures/Suppl_Enrichm_Fct3.pdf&quot;, width = 183, height = 183, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(plot.enriched.go.bp.factor3_50, filename = &quot;output/paper_figures/Suppl_Enrichm_Fct3.png&quot;, width = 183, height = 183, units = &quot;mm&quot;,  dpi = 300)
plot.enriched.go.bp.factor3_50</code></pre>
<p><img src="figure/MOFAaIg.Rmd/Suppl-1.png" width="691.2" style="display: block; margin: auto;" /> <em>Supplementary Figure.</em> Top 50 enriched gene-sets in postive loadings factor 3.</p>
</div>
<div id="suppl-pjak1-high-vs-low" class="section level3">
<h3>Suppl pJak1 High vs low</h3>
<pre class="r"><code>proteindata_counts &lt;- as.data.frame(t(seu_combined_selectsamples@assays$PROT@scale.data)) %&gt;%
  mutate(sample = rownames(t(seu_combined_selectsamples@assays$PROT@scale.data)))


metadata.all &lt;- as.data.frame(seu_combined_selectsamples@meta.data) %&gt;%
  mutate(sample = rownames((seu_combined_selectsamples@meta.data)))

proteindata_counts &lt;- left_join(proteindata_counts, metadata.all)



toppJAK1 &lt;- proteindata_counts %&gt;%
  group_by(time) %&gt;%
  top_frac(wt = `p-JAK1`, n = 0.05)

bottompJAK1 &lt;- proteindata_counts %&gt;%
  group_by(time) %&gt;%
  top_frac(wt = `p-JAK1`, n = -0.05)

proteindata_counts &lt;- proteindata_counts %&gt;%
  mutate(highlowpJAK1 = ifelse(sample %in% toppJAK1$sample, &quot;p-JAK1 high&quot;, ifelse(sample %in% bottompJAK1$sample, &quot;p-JAK1 low&quot;,&quot;middle&quot;)))

addmeta &lt;- proteindata_counts[,c(&quot;highlowpJAK1&quot;, &quot;sample&quot;)]
rownames(addmeta) &lt;- proteindata_counts$sample

seu_combined_selectsamples &lt;- AddMetaData(seu_combined_selectsamples, addmeta)

seu.JAK1.180 &lt;- subset(seu_combined_selectsamples, condition == &quot;180.aIg.contr&quot; &amp; highlowpJAK1 != &quot;middle&quot;)

seu.JAK1.180 &lt;- SetIdent(seu.JAK1.180, value = &quot;highlowpJAK1&quot;)</code></pre>
<pre class="r"><code># Find differentially expressed features between CD14+ and FCGR3A+ Monocytes
markers.180 &lt;- FindMarkers(seu.JAK1.180,ident.1 = &quot;p-JAK1 high&quot;, ident.2 = &quot;p-JAK1 low&quot;, assay = &quot;PROT&quot;, slot = &quot;scale.data&quot;, logfc.threshold = 0, return.thresh = 1, only.pos = F)
# view results
#markers.180 &lt;- filter(markers.180, cluster == &quot;p-JAK1 high&quot;)
#markers.180


# Find differentially expressed features between CD14+ and FCGR3A+ Monocytes
markers.180.RNA &lt;-  FindAllMarkers(seu.JAK1.180,assay = &quot;RNA&quot;, slot = &quot;data&quot;, logfc.threshold = 0.3, return.thresh = 0.01, only.pos = T,min.pct = 0.1)
markers.180.RNA &lt;-   FindMarkers(seu.JAK1.180,ident.1 = &quot;p-JAK1 high&quot;, ident.2 = &quot;p-JAK1 low&quot;, assay = &quot;SCT.RNA&quot;, slot = &quot;scale.data&quot;, logfc.threshold = 0, return.thresh = 1, only.pos = F)
# view results
#markers.180.RNA

markers.180$protein &lt;-rownames(markers.180)
# add a column of NAs
markers.180$diffexpressed &lt;- &quot;NO&quot;
# if log2Foldchange &gt; 0.6 and pvalue &lt; 0.05, set as &quot;UP&quot; 
markers.180$diffexpressed[markers.180$avg_diff &gt; 0.25 &amp; markers.180$p_val_adj &lt; 0.01] &lt;- &quot;UP&quot;
# if log2Foldchange &lt; -0.6 and pvalue &lt; 0.05, set as &quot;DOWN&quot;
markers.180$diffexpressed[markers.180$avg_diff  &lt; -0.25 &amp; markers.180$p_val_adj &lt; 0.01] &lt;- &quot;DOWN&quot;

mycolors &lt;- c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)
names(mycolors) &lt;- c(&quot;DOWN&quot;, &quot;UP&quot;, &quot;NO&quot;)


markers.180$delabel &lt;- NA
markers.180$delabel[markers.180$diffexpressed != &quot;NO&quot;] &lt;- markers.180$protein[markers.180$diffexpressed != &quot;NO&quot;]

# Finally, we can organize the labels nicely using the &quot;ggrepel&quot; package and the geom_text_repel() function
# load library

# plot adding up all layers we have seen so far
plot.vulcano.180min &lt;- ggplot(data=markers.180, aes(x=avg_diff , y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=0.5) + 
        theme_minimal() +
        geom_text_repel(size=2.2) +
        scale_color_manual(values=c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)) +
        geom_vline(xintercept=c(-0.25, 0.25), col=&quot;red&quot;) +
        geom_hline(yintercept=-log10(0.01), col=&quot;red&quot;) +
  labs(x = expression(&quot;Log&quot;[2]*&quot; Fold Change&quot;), y = expression(&quot;-log&quot;[10]*&quot; adjusted p-value&quot;), title = &quot;p-JAK1 high vs p-JAK1 low (t = 180 min)&quot;) &amp;
  add.textsize

sign.markers180 &lt;- markers.180$protein[markers.180$avg_diff &gt; 0.25 &amp; markers.180$p_val_adj &lt; 0.01 | markers.180$avg_diff  &lt; -0.25 &amp; markers.180$p_val_adj &lt; 0.01] 

plot.vln.180min &lt;- VlnPlot(seu.JAK1.180,assay = &quot;PROT&quot;,slot = &quot;scale.data&quot;, features =  sign.markers180, group.by = &quot;highlowpJAK1&quot;,ncol = 6, pt.size = 0.5) &amp;
  add.textsize

## RNA
markers.180.RNA$protein &lt;-rownames(markers.180.RNA)
# add a column of NAs
markers.180.RNA$diffexpressed &lt;- &quot;NO&quot;
# if log2Foldchange &gt; 0.6 and pvalue &lt; 0.05, set as &quot;UP&quot; 
markers.180.RNA$diffexpressed[markers.180.RNA$avg_diff &gt; 0.25 &amp; markers.180.RNA$p_val &lt; 0.05] &lt;- &quot;UP&quot;
# if log2Foldchange &lt; -0.6 and pvalue &lt; 0.05, set as &quot;DOWN&quot;
markers.180.RNA$diffexpressed[markers.180.RNA$avg_diff &lt; -0.25 &amp; markers.180.RNA$p_val &lt; 0.05] &lt;- &quot;DOWN&quot;

mycolors &lt;- c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)
names(mycolors) &lt;- c(&quot;DOWN&quot;, &quot;UP&quot;, &quot;NO&quot;)


markers.180.RNA$delabel &lt;- NA
markers.180.RNA$delabel[markers.180.RNA$diffexpressed != &quot;NO&quot;] &lt;- markers.180.RNA$protein[markers.180.RNA$diffexpressed != &quot;NO&quot;]

# plot adding up all layers we have seen so far
plot.vulcano.180min.RNA &lt;- ggplot(data=markers.180.RNA, aes(x=avg_diff, y=-log10(p_val), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        geom_text_repel(size=2.2) +
        scale_color_manual(values=c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)) +
        geom_vline(xintercept=c(-0.25, 0.25), col=&quot;red&quot;) +
        geom_hline(yintercept=-log10(0.05), col=&quot;red&quot;) +
  labs(x = expression(&quot;Log&quot;[2]*&quot; Fold Change&quot;), y = expression(&quot;-log&quot;[10]*&quot; p-value&quot;), title = &quot;p-JAK1 high vs p-JAK1 low (t = 180 min)&quot;) &amp;
  add.textsize

sign.markers180.RNA &lt;- markers.180.RNA$protein[markers.180.RNA$avg_diff &gt; 0.25 &amp; markers.180.RNA$p_val &lt; 0.05] 

plot.vln.180min.RNA &lt;- VlnPlot(seu.JAK1.180,assay = &quot;RNA&quot;, features =  sign.markers180.RNA[1:20], group.by = &quot;highlowpJAK1&quot;,ncol = 10) &amp;
  add.textsize

plot_180min &lt;- plot_grid(plot.vulcano.180min, plot.vln.180min, labels = panellabels[c(5,6)], label_size = 10, ncol = 2, rel_widths = c(1,2))

## 6 minutes
seu.JAK1.006 &lt;- subset(seu_combined_selectsamples, condition == &quot;006.aIg.contr&quot; &amp; highlowpJAK1 != &quot;middle&quot;)

seu.JAK1.006 &lt;- SetIdent(seu.JAK1.006, value = &quot;highlowpJAK1&quot;)

# Find differentially expressed features between CD14+ and FCGR3A+ Monocytes
markers.006 &lt;- FindMarkers(seu.JAK1.006,ident.1 = &quot;p-JAK1 high&quot;, ident.2 = &quot;p-JAK1 low&quot;, assay = &quot;PROT&quot;, slot = &quot;scale.data&quot;, logfc.threshold = 0, return.thresh = 1, only.pos = F)
# view results
#markers.006 &lt;- filter(markers.006, cluster == &quot;p-JAK1 high&quot;)
#markers.006


# Find differentially expressed features between CD14+ and FCGR3A+ Monocytes
markers.006.RNA &lt;-  FindAllMarkers(seu.JAK1.006,assay = &quot;RNA&quot;, slot = &quot;data&quot;, logfc.threshold = 0.3, return.thresh = 0.01, only.pos = T,min.pct = 0.1)
markers.006.RNA &lt;-   FindMarkers(seu.JAK1.006,ident.1 = &quot;p-JAK1 high&quot;, ident.2 = &quot;p-JAK1 low&quot;, assay = &quot;SCT.RNA&quot;, slot = &quot;scale.data&quot;, logfc.threshold = 0, return.thresh = 1, only.pos = F)
# view results
#markers.006.RNA

markers.006$protein &lt;-rownames(markers.006)
# add a column of NAs
markers.006$diffexpressed &lt;- &quot;NO&quot;
# if log2Foldchange &gt; 0.6 and pvalue &lt; 0.05, set as &quot;UP&quot; 
markers.006$diffexpressed[markers.006$avg_diff &gt; 0.25 &amp; markers.006$p_val_adj &lt; 0.01] &lt;- &quot;UP&quot;
# if log2Foldchange &lt; -0.6 and pvalue &lt; 0.05, set as &quot;DOWN&quot;
markers.006$diffexpressed[markers.006$avg_diff  &lt; -0.25 &amp; markers.006$p_val_adj &lt; 0.01] &lt;- &quot;DOWN&quot;

mycolors &lt;- c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)
names(mycolors) &lt;- c(&quot;DOWN&quot;, &quot;UP&quot;, &quot;NO&quot;)


markers.006$delabel &lt;- NA
markers.006$delabel[markers.006$diffexpressed != &quot;NO&quot;] &lt;- markers.006$protein[markers.006$diffexpressed != &quot;NO&quot;]

# Finally, we can organize the labels nicely using the &quot;ggrepel&quot; package and the geom_text_repel() function
# load library
# plot adding up all layers we have seen so far
plot.vulcano.006min &lt;- ggplot(data=markers.006, aes(x=avg_diff , y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=0.5) + 
        theme_minimal() +
        geom_text_repel(size=2.2) +
        scale_color_manual(values=c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)) +
        geom_vline(xintercept=c(-0.25, 0.25), col=&quot;red&quot;) +
        geom_hline(yintercept=-log10(0.01), col=&quot;red&quot;) +
  labs(x = expression(&quot;Log&quot;[2]*&quot; Fold Change&quot;), y = expression(&quot;-log&quot;[10]*&quot; adjusted p-value&quot;), title = &quot;p-JAK1 high vs p-JAK1 low (t = 006 min)&quot;) &amp;
  add.textsize

sign.markers006 &lt;- markers.006$protein[markers.006$avg_diff &gt; 0.25 &amp; markers.006$p_val_adj &lt; 0.01 | markers.006$avg_diff  &lt; -0.25 &amp; markers.006$p_val_adj &lt; 0.01] 

plot.vln.006min &lt;- VlnPlot(seu.JAK1.006,assay = &quot;PROT&quot;,slot = &quot;scale.data&quot;, features =  sign.markers006, group.by = &quot;highlowpJAK1&quot;,ncol = 6, pt.size = 0.5) &amp;
  add.textsize

markers.006.RNA$protein &lt;-rownames(markers.006.RNA)
# add a column of NAs
markers.006.RNA$diffexpressed &lt;- &quot;NO&quot;
# if log2Foldchange &gt; 0.6 and pvalue &lt; 0.05, set as &quot;UP&quot; 
markers.006.RNA$diffexpressed[markers.006.RNA$avg_diff &gt; 0.25 &amp; markers.006.RNA$p_val &lt; 0.05] &lt;- &quot;UP&quot;
# if log2Foldchange &lt; -0.6 and pvalue &lt; 0.05, set as &quot;DOWN&quot;
markers.006.RNA$diffexpressed[markers.006.RNA$avg_diff &lt; -0.25 &amp; markers.006.RNA$p_val &lt; 0.05] &lt;- &quot;DOWN&quot;

mycolors &lt;- c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)
names(mycolors) &lt;- c(&quot;DOWN&quot;, &quot;UP&quot;, &quot;NO&quot;)


markers.006.RNA$delabel &lt;- NA
markers.006.RNA$delabel[markers.006.RNA$diffexpressed != &quot;NO&quot;] &lt;- markers.006.RNA$protein[markers.006.RNA$diffexpressed != &quot;NO&quot;]

# Finally, we can organize the labels nicely using the &quot;ggrepel&quot; package and the geom_text_repel() function
# load library
# plot adding up all layers we have seen so far
plot.vulcano.006min.RNA &lt;- ggplot(data=markers.006.RNA, aes(x=avg_diff, y=-log10(p_val), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        geom_text_repel(size=2.2) +
        scale_color_manual(values=c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)) +
        geom_vline(xintercept=c(-0.25, 0.25), col=&quot;red&quot;) +
        geom_hline(yintercept=-log10(0.05), col=&quot;red&quot;) +
  labs(x = expression(&quot;Log&quot;[2]*&quot; Fold Change&quot;), y = expression(&quot;-log&quot;[10]*&quot; p-value&quot;), title = &quot;p-JAK1 high vs p-JAK1 low (t = 006 min)&quot;) &amp;
  add.textsize

sign.markers006.RNA &lt;- markers.006.RNA$protein[markers.006.RNA$avg_diff &gt; 0.25 &amp; markers.006.RNA$p_val &lt; 0.05] 

plot.vln.006min.RNA &lt;- VlnPlot(seu.JAK1.006,assay = &quot;RNA&quot;, features =  sign.markers006.RNA[1:20], group.by = &quot;highlowpJAK1&quot;,ncol = 10) &amp;
  add.textsize

plot_006min &lt;- plot_grid(plot.vulcano.006min, plot.vln.006min, labels = panellabels[c(3,4)], label_size = 10, ncol = 2, rel_widths = c(1,2))

## 2 minutes
seu.JAK1.002 &lt;- subset(seu_combined_selectsamples, condition == &quot;002.aIg.contr&quot; &amp; highlowpJAK1 != &quot;middle&quot;)

seu.JAK1.002 &lt;- SetIdent(seu.JAK1.002, value = &quot;highlowpJAK1&quot;)

# Find differentially expressed features between CD14+ and FCGR3A+ Monocytes
markers.002 &lt;- FindMarkers(seu.JAK1.002,ident.1 = &quot;p-JAK1 high&quot;, ident.2 = &quot;p-JAK1 low&quot;, assay = &quot;PROT&quot;, slot = &quot;scale.data&quot;, logfc.threshold = 0, return.thresh = 1, only.pos = F)
# view results
#markers.002 &lt;- filter(markers.002, cluster == &quot;p-JAK1 high&quot;)
#markers.002


# Find differentially expressed features between CD14+ and FCGR3A+ Monocytes
markers.002.RNA &lt;-  FindAllMarkers(seu.JAK1.002,assay = &quot;RNA&quot;, slot = &quot;data&quot;, logfc.threshold = 0.3, return.thresh = 0.01, only.pos = T,min.pct = 0.1)
markers.002.RNA &lt;-   FindMarkers(seu.JAK1.002,ident.1 = &quot;p-JAK1 high&quot;, ident.2 = &quot;p-JAK1 low&quot;, assay = &quot;SCT.RNA&quot;, slot = &quot;scale.data&quot;, logfc.threshold = 0, return.thresh = 1, only.pos = F)
# view results
#markers.002.RNA

markers.002$protein &lt;-rownames(markers.002)
# add a column of NAs
markers.002$diffexpressed &lt;- &quot;NO&quot;
# if log2Foldchange &gt; 0.6 and pvalue &lt; 0.05, set as &quot;UP&quot; 
markers.002$diffexpressed[markers.002$avg_diff &gt; 0.25 &amp; markers.002$p_val_adj &lt; 0.01] &lt;- &quot;UP&quot;
# if log2Foldchange &lt; -0.6 and pvalue &lt; 0.05, set as &quot;DOWN&quot;
markers.002$diffexpressed[markers.002$avg_diff  &lt; -0.25 &amp; markers.002$p_val_adj &lt; 0.01] &lt;- &quot;DOWN&quot;

mycolors &lt;- c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)
names(mycolors) &lt;- c(&quot;DOWN&quot;, &quot;UP&quot;, &quot;NO&quot;)


markers.002$delabel &lt;- NA
markers.002$delabel[markers.002$diffexpressed != &quot;NO&quot;] &lt;- markers.002$protein[markers.002$diffexpressed != &quot;NO&quot;]

plot.vulcano.002min &lt;- ggplot(data=markers.002, aes(x=avg_diff , y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=0.5) + 
        theme_minimal() +
        geom_text_repel(size=2.2) +
        scale_color_manual(values=c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)) +
        geom_vline(xintercept=c(-0.25, 0.25), col=&quot;red&quot;) +
        geom_hline(yintercept=-log10(0.01), col=&quot;red&quot;) +
  labs(x = expression(&quot;Log&quot;[2]*&quot; Fold Change&quot;), y = expression(&quot;-log&quot;[10]*&quot; adjusted p-value&quot;), title = &quot;p-JAK1 high vs p-JAK1 low (t = 002 min)&quot;) &amp;
  add.textsize

sign.markers002 &lt;- markers.002$protein[markers.002$avg_diff &gt; 0.25 &amp; markers.002$p_val_adj &lt; 0.01 | markers.002$avg_diff  &lt; -0.25 &amp; markers.002$p_val_adj &lt; 0.01] 

plot.vln.002min &lt;- VlnPlot(seu.JAK1.002,assay = &quot;PROT&quot;,slot = &quot;scale.data&quot;, features =  sign.markers002, group.by = &quot;highlowpJAK1&quot;,ncol = 6, pt.size = 0.5) &amp;
  add.textsize

markers.002.RNA$protein &lt;-rownames(markers.002.RNA)
# add a column of NAs
markers.002.RNA$diffexpressed &lt;- &quot;NO&quot;
# if log2Foldchange &gt; 0.6 and pvalue &lt; 0.05, set as &quot;UP&quot; 
markers.002.RNA$diffexpressed[markers.002.RNA$avg_diff &gt; 0.25 &amp; markers.002.RNA$p_val &lt; 0.05] &lt;- &quot;UP&quot;
# if log2Foldchange &lt; -0.6 and pvalue &lt; 0.05, set as &quot;DOWN&quot;
markers.002.RNA$diffexpressed[markers.002.RNA$avg_diff &lt; -0.25 &amp; markers.002.RNA$p_val &lt; 0.05] &lt;- &quot;DOWN&quot;

mycolors &lt;- c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)
names(mycolors) &lt;- c(&quot;DOWN&quot;, &quot;UP&quot;, &quot;NO&quot;)


markers.002.RNA$delabel &lt;- NA
markers.002.RNA$delabel[markers.002.RNA$diffexpressed != &quot;NO&quot;] &lt;- markers.002.RNA$protein[markers.002.RNA$diffexpressed != &quot;NO&quot;]

plot.vulcano.002min.RNA &lt;- ggplot(data=markers.002.RNA, aes(x=avg_diff, y=-log10(p_val), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        geom_text_repel(size=2.2) +
        scale_color_manual(values=c(&quot;blue&quot;, &quot;red&quot;, &quot;black&quot;)) +
        geom_vline(xintercept=c(-0.25, 0.25), col=&quot;red&quot;) +
        geom_hline(yintercept=-log10(0.05), col=&quot;red&quot;) +
  labs(x = expression(&quot;Log&quot;[2]*&quot; Fold Change&quot;), y = expression(&quot;-log&quot;[10]*&quot; p-value&quot;), title = &quot;p-JAK1 high vs p-JAK1 low (t = 002 min)&quot;) &amp;
  add.textsize

sign.markers002.RNA &lt;- markers.002.RNA$protein[markers.002.RNA$avg_diff &gt; 0.25 &amp; markers.002.RNA$p_val &lt; 0.05] 

plot.vln.002min.RNA &lt;- VlnPlot(seu.JAK1.002,assay = &quot;RNA&quot;, features =  sign.markers002.RNA[1:20], group.by = &quot;highlowpJAK1&quot;,ncol = 10) &amp;
  add.textsize

plot_002min &lt;- plot_grid(plot.vulcano.002min, plot.vln.002min, labels = panellabels[c(1,2)], label_size = 10, ncol = 2, rel_widths = c(1,2))</code></pre>
<pre class="r"><code>plot_all &lt;- plot_grid(plot_002min,plot_006min,plot_180min , labels =c(&quot;&quot;, &quot;&quot;, &quot;&quot;), ncol = 1, rel_heights = c(1,1,1))

ggsave(plot_all, filename = &quot;output/paper_figures/Suppl_pJAK1_highlow.pdf&quot;, width = 183, height = 140, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(plot_all, filename = &quot;output/paper_figures/Suppl_pJAK1_highlow.png&quot;, width = 183, height = 140, units = &quot;mm&quot;,  dpi = 300)

plot_all</code></pre>
<p><img src="figure/MOFAaIg.Rmd/plot_all-1.png" width="691.2" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   
[3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.1252    

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] png_0.1-7                   forcats_0.5.1              
 [3] clusterProfiler_3.18.1      clusterProfiler.dplyr_0.0.2
 [5] enrichplot_1.10.2           org.Hs.eg.db_3.12.0        
 [7] AnnotationDbi_1.52.0        IRanges_2.24.1             
 [9] S4Vectors_0.28.1            Biobase_2.50.0             
[11] BiocGenerics_0.36.0         ggridges_0.5.3             
[13] cowplot_1.1.1               ggtext_0.1.1               
[15] ggplotify_0.0.5             ggcorrplot_0.1.3           
[17] ggrepel_0.9.1               ggpubr_0.4.0               
[19] scico_1.2.0                 MOFA2_1.1.17               
[21] extrafont_0.17              patchwork_1.1.1            
[23] RColorBrewer_1.1-2          viridis_0.5.1              
[25] viridisLite_0.3.0           ggsci_2.9                  
[27] sctransform_0.3.2           ggthemes_4.2.4             
[29] matrixStats_0.57.0          kableExtra_1.3.1           
[31] gridExtra_2.3               SeuratObject_4.0.0         
[33] Seurat_4.0.0                ggplot2_3.3.3              
[35] scales_1.1.1                tidyr_1.1.2                
[37] dplyr_1.0.3                 stringr_1.4.0              
[39] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.2       scattermore_0.7      bit64_4.0.5         
  [4] knitr_1.31           irlba_2.3.3          DelayedArray_0.16.1 
  [7] data.table_1.13.6    rpart_4.1-15         generics_0.1.0      
 [10] RSQLite_2.2.3        shadowtext_0.0.7     RANN_2.6.1          
 [13] future_1.21.0        bit_4.0.4            spatstat.data_2.1-0 
 [16] webshot_0.5.2        xml2_1.3.2           httpuv_1.5.5        
 [19] assertthat_0.2.1     xfun_0.23            hms_1.0.0           
 [22] evaluate_0.14        promises_1.1.1       readxl_1.3.1        
 [25] tmvnsim_1.0-2        igraph_1.2.6         DBI_1.1.1           
 [28] htmlwidgets_1.5.3    purrr_0.3.4          ellipsis_0.3.1      
 [31] RSpectra_0.16-0      corrplot_0.84        backports_1.2.1     
 [34] markdown_1.1         deldir_0.2-10        MatrixGenerics_1.2.0
 [37] vctrs_0.3.6          ROCR_1.0-11          abind_1.4-5         
 [40] cachem_1.0.1         withr_2.4.1          ggforce_0.3.2       
 [43] mnormt_2.0.2         goftest_1.2-2        cluster_2.1.0       
 [46] DOSE_3.16.0          lazyeval_0.2.2       crayon_1.3.4        
 [49] basilisk.utils_1.2.1 pkgconfig_2.0.3      labeling_0.4.2      
 [52] tweenr_1.0.1         nlme_3.1-149         rlang_0.4.10        
 [55] globals_0.14.0       lifecycle_0.2.0      miniUI_0.1.1.1      
 [58] downloader_0.4       filelock_1.0.2       extrafontdb_1.0     
 [61] cellranger_1.1.0     rprojroot_2.0.2      polyclip_1.10-0     
 [64] lmtest_0.9-38        Matrix_1.2-18        carData_3.0-4       
 [67] Rhdf5lib_1.12.1      zoo_1.8-8            whisker_0.4         
 [70] pheatmap_1.0.12      KernSmooth_2.23-17   rhdf5filters_1.2.0  
 [73] blob_1.2.1           qvalue_2.22.0        parallelly_1.23.0   
 [76] rstatix_0.6.0        gridGraphics_0.5-1   ggsignif_0.6.0      
 [79] memoise_2.0.0        magrittr_2.0.1       plyr_1.8.6          
 [82] ica_1.0-2            compiler_4.0.3       scatterpie_0.1.5    
 [85] fitdistrplus_1.1-3   listenv_0.8.0        pbapply_1.4-3       
 [88] MASS_7.3-53          mgcv_1.8-33          tidyselect_1.1.0    
 [91] stringi_1.5.3        highr_0.8            yaml_2.2.1          
 [94] GOSemSim_2.16.1      fastmatch_1.1-0      tools_4.0.3         
 [97] future.apply_1.7.0   rio_0.5.16           rstudioapi_0.13     
[100] foreign_0.8-80       git2r_0.28.0         farver_2.0.3        
[103] Rtsne_0.15           ggraph_2.0.5         digest_0.6.27       
[106] rvcheck_0.1.8        BiocManager_1.30.10  shiny_1.6.0         
[109] Rcpp_1.0.6           gridtext_0.1.4       car_3.0-10          
[112] broom_0.7.3          later_1.1.0.1        RcppAnnoy_0.0.18    
[115] httr_1.4.2           psych_2.0.12         colorspace_2.0-0    
[118] rvest_0.3.6          fs_1.5.0             tensor_1.5          
[121] reticulate_1.18      splines_4.0.3        uwot_0.1.10         
[124] spatstat.utils_2.1-0 graphlayouts_0.7.1   basilisk_1.2.1      
[127] plotly_4.9.3         xtable_1.8-4         jsonlite_1.7.2      
[130] spatstat_1.64-1      tidygraph_1.2.0      R6_2.5.0            
[133] pillar_1.4.7         htmltools_0.5.1.1    mime_0.9            
[136] glue_1.4.2           fastmap_1.1.0        BiocParallel_1.24.1 
[139] codetools_0.2-16     fgsea_1.16.0         lattice_0.20-41     
[142] tibble_3.0.5         curl_4.3             leiden_0.3.7        
[145] zip_2.1.1            GO.db_3.12.1         openxlsx_4.2.3      
[148] Rttf2pt1_1.3.8       limma_3.46.0         survival_3.2-7      
[151] rmarkdown_2.6        munsell_0.5.0        DO.db_2.9           
[154] rhdf5_2.34.0         HDF5Array_1.18.0     haven_2.3.1         
[157] reshape2_1.4.4       gtable_0.3.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
