<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Quality control and normalization</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manuscript QuRIE-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="QC.html">QC</a>
</li>
<li>
  <a href="MOFAaIg.html">MOFA aIg</a>
</li>
<li>
  <a href="MOFAibru.html">MOFA ibru</a>
</li>
<li>
  <a href="license.html">License and citation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Quality control and normalization</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>QuRIE-seq_manuscript/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201117code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201117)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201117code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201117)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomvanbuggenumQuRIEseqmanuscripttree4f4828dc0b105e22d522002a43d0f6b17d11bacdtargetblank4f4828da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/tree/4f4828dc0b105e22d522002a43d0f6b17d11bacd" target="_blank">4f4828d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomvanbuggenumQuRIEseqmanuscripttree4f4828dc0b105e22d522002a43d0f6b17d11bacdtargetblank4f4828da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/tree/4f4828dc0b105e22d522002a43d0f6b17d11bacd" target="_blank">4f4828d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/raw/
    Ignored:    output/MOFA_aIg.hdf5
    Ignored:    output/MOFA_ibru.hdf5
    Ignored:    output/data_weights_prot_fact1and3.csv
    Ignored:    output/metadata_Ricard.csv

Unstaged changes:
    Modified:   analysis/revisions.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/QC.Rmd</code>) and HTML (<code>docs/QC.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/4f4828dc0b105e22d522002a43d0f6b17d11bacd/analysis/QC.Rmd" target="_blank">4f4828d</a>
</td>
<td>
jessievb
</td>
<td>
2021-07-08
</td>
<td>
update header and small edits html output
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/c51a9e773353e491548a8cd9e2985235d0b8cce8/docs/QC.html" target="_blank">c51a9e7</a>
</td>
<td>
jessievb
</td>
<td>
2021-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/47ecaed94474bfd9d71d03049c448dd3457480c8/docs/QC.html" target="_blank">47ecaed</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/14c0ebf2831e464136125855b1d78cfbf533fffe/analysis/QC.Rmd" target="_blank">14c0ebf</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
small update documentation github page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/eda7e9302a1081b93794d5ca9868d9af7adbc737/docs/QC.html" target="_blank">eda7e93</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/0d971e2a6ef8f3824d925655d662b7ceeacec605/analysis/QC.Rmd" target="_blank">0d971e2</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-24
</td>
<td>
minor change docs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/064f105e8eb70252d7c635dbc87c4f5c1ca18c4e/docs/QC.html" target="_blank">064f105</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/420f8973a6de641d96917dda9684f5994b8f5f18/analysis/QC.Rmd" target="_blank">420f897</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-20
</td>
<td>
update gene treshold
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/1f83dd100d4b2069c03c893fb546676351e5619f/docs/QC.html" target="_blank">1f83dd1</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/5d80bc60e41940bb4c7d346d5da4aae483246ba3/analysis/QC.Rmd" target="_blank">5d80bc6</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-19
</td>
<td>
correct tresholds manuscript
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/2c3e4a119dcf91fab22e312f9c4b928a09bee26f/docs/QC.html" target="_blank">2c3e4a1</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/67539d569507cb2f30d2188677021e9b34659fa7/analysis/QC.Rmd" target="_blank">67539d5</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-18
</td>
<td>
QC - filter and normalize datasets and split to subset of data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/QuRIE-seq_manuscript/2ffe38d8ad99572f5b5e0fdde94cbd8db50659bf/docs/QC.html" target="_blank">2ffe38d</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/4487750fd014ae3af0bfaa3cd6d581cf1ac3392c/analysis/QC.Rmd" target="_blank">4487750</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-17
</td>
<td>
basis set-up of page display
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/QuRIE-seq_manuscript/blob/d9372d117842a51e92f9ba0293d6e92c8e9b29c1/analysis/QC.Rmd" target="_blank">d9372d1</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2020-11-17
</td>
<td>
Publish the initial files for this website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>source(&quot;code/load_packages.R&quot;)
source(&quot;code/plot_QC_function.R&quot;) #change settings in script to get different layout per subpanel for this figure
dir.create(&quot;output/paper_figures&quot;) # location where paper figures are stored</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The QuRIE-seq data contains single-cell transcriptomic and proteomic data of BJAB cells with 9 different durations of aIg stimulation, and 3 additional timepoints with ibrutinib drug inhibition. Script below extracts all metadata (sequencing sample names, adds metadata info per sample (prot or RNA library)), and reads all data tables into R. The metadata table is saved in output folder: “output/metadata.csv”</p>
<p>For QC and filtering, cells with matching RNA and protein information are used to create a Seurat object (settings RNA: min.cells = 100, min.features = 100; proteins added as additional modality PROT). Several QC stats are calculated, and the object is saved in: “output/seu_combined_raw.rds”</p>
<pre class="r"><code>myfiles &lt;- list.files(path=&quot;output/&quot;, pattern = &quot;.rds$&quot;)
## only read all raw files and create raw combined table if not done yet. Speeds up generation of html file
if(&quot;seu_combined_raw.rds&quot; %in%  myfiles){seu_combined &lt;- readRDS(&quot;output/seu_combined_raw.rds&quot;)} else { 
  source(&quot;code/Import_and_create_seuratObj.R&quot;)
}</code></pre>
</div>
<div id="before-qc-dataset-properties" class="section level2">
<h2>Before QC dataset properties</h2>
<p>Seurat object:</p>
<pre class="r"><code>seu_combined</code></pre>
<pre><code>An object of class Seurat 
8452 features across 7449 samples within 2 assays 
Active assay: RNA (8372 features, 0 variable features)
 1 other assay present: PROT</code></pre>
<p><strong>Table</strong> Overview of per sample properties.</p>
<pre class="r"><code>kable(seu_combined@meta.data %&gt;% 
        group_by(condition) %&gt;% 
        summarise(`Total number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  )) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
condition
</th>
<th style="text-align:right;">
Total number of cells
</th>
<th style="text-align:right;">
Median counts RNA
</th>
<th style="text-align:right;">
Median Number genes
</th>
<th style="text-align:right;">
Median Mitochondrial counts (Median %)
</th>
<th style="text-align:right;">
Ribosomal counts (Median %)
</th>
<th style="text-align:right;">
Median counts PROT
</th>
<th style="text-align:right;">
Number proteins
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
000.aIg.contr
</td>
<td style="text-align:right;">
688
</td>
<td style="text-align:right;">
1364
</td>
<td style="text-align:right;">
947
</td>
<td style="text-align:right;">
7.32
</td>
<td style="text-align:right;">
6.76
</td>
<td style="text-align:right;">
1108
</td>
<td style="text-align:right;">
61
</td>
</tr>
<tr>
<td style="text-align:left;">
002.aIg.contr
</td>
<td style="text-align:right;">
958
</td>
<td style="text-align:right;">
772
</td>
<td style="text-align:right;">
585
</td>
<td style="text-align:right;">
7.81
</td>
<td style="text-align:right;">
6.83
</td>
<td style="text-align:right;">
984
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
004.aIg.contr
</td>
<td style="text-align:right;">
545
</td>
<td style="text-align:right;">
829
</td>
<td style="text-align:right;">
619
</td>
<td style="text-align:right;">
8.04
</td>
<td style="text-align:right;">
6.06
</td>
<td style="text-align:right;">
1035
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.contr
</td>
<td style="text-align:right;">
820
</td>
<td style="text-align:right;">
1047
</td>
<td style="text-align:right;">
762
</td>
<td style="text-align:right;">
8.37
</td>
<td style="text-align:right;">
6.93
</td>
<td style="text-align:right;">
720
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.ibr
</td>
<td style="text-align:right;">
1148
</td>
<td style="text-align:right;">
459
</td>
<td style="text-align:right;">
354
</td>
<td style="text-align:right;">
11.14
</td>
<td style="text-align:right;">
6.10
</td>
<td style="text-align:right;">
756
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
060.aIg.contr
</td>
<td style="text-align:right;">
879
</td>
<td style="text-align:right;">
636
</td>
<td style="text-align:right;">
489
</td>
<td style="text-align:right;">
8.84
</td>
<td style="text-align:right;">
6.71
</td>
<td style="text-align:right;">
891
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.contr
</td>
<td style="text-align:right;">
1121
</td>
<td style="text-align:right;">
710
</td>
<td style="text-align:right;">
548
</td>
<td style="text-align:right;">
7.43
</td>
<td style="text-align:right;">
6.85
</td>
<td style="text-align:right;">
867
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.ibr
</td>
<td style="text-align:right;">
1290
</td>
<td style="text-align:right;">
735
</td>
<td style="text-align:right;">
572
</td>
<td style="text-align:right;">
6.37
</td>
<td style="text-align:right;">
6.97
</td>
<td style="text-align:right;">
808
</td>
<td style="text-align:right;">
57
</td>
</tr>
</tbody>
</table>
<p><strong>Table</strong> Overview of full dataset properties.</p>
<pre class="r"><code>kable(seu_combined@meta.data %&gt;% 
        summarise(`Number of cells` = round(n(),0),
          `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  ) %&gt;%
        t()) %&gt;%

  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
7449.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts RNA
</td>
<td style="text-align:right;">
727.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Number genes
</td>
<td style="text-align:right;">
556.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Mitochondrial counts (Median %)
</td>
<td style="text-align:right;">
8.05
</td>
</tr>
<tr>
<td style="text-align:left;">
Ribosomal counts (Median %)
</td>
<td style="text-align:right;">
6.70
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts PROT
</td>
<td style="text-align:right;">
863.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Number proteins
</td>
<td style="text-align:right;">
57.00
</td>
</tr>
</tbody>
</table>
</div>
<div id="quality-control-filtering" class="section level2">
<h2>Quality Control &amp; filtering</h2>
<pre class="r"><code>plot_RNA_nCount &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;nCount_RNA&quot;, 
                                 ytext = &quot;Total UMI counts per cell&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Keep cells &lt; 4000 RNA counts&quot;,
                                 colorviolin = &quot;dodgerblue2&quot; ) + 
                        geom_hline(yintercept = 4000, size = 0.3) +
                        theme(axis.title.x = element_blank())

plot_RNA_ngenes &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;nFeature_RNA&quot;, 
                                 ytext = &quot;Total genes per cell&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Keep cells &gt;150 genes&quot;,
                                 colorviolin = &quot;dodgerblue2&quot; ) + 
                        geom_hline(yintercept = 150, size = 0.3) +
                        theme(axis.title.x = element_blank())
  
plot_percent.mt &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;percent.mt&quot;, 
                                 ytext = &quot;% Mitochondrial counts&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Keep cells &lt; 15 % mitochondrial genecounts&quot;,
                                 colorviolin = &quot;dodgerblue2&quot; ) +
                        geom_hline(yintercept = 15, color = &quot;black&quot;, size = 0.3) +
                        theme(axis.title.x = element_blank())

plot_percent.rb &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;percent.rb&quot;, 
                                 ytext = &quot;% Ribosomal counts&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Stable % ribosomal counts over time&quot;,
                                 colorviolin = &quot;dodgerblue2&quot; ) +
                        theme(axis.title.x = element_blank())
  
  
plot_PROT_nCount &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;nCount_PROT&quot;, 
                                 ytext =  &quot;Total UMI counts per cell&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Keep cells &lt; 3000 PROT counts&quot;,
                                 colorviolin = &quot;deeppink3&quot; ) +
                        geom_hline(yintercept = 3000, size = 0.3) +
                        theme(axis.title.x = element_blank())

plot_PROT_nproteins &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;nFeature_PROT&quot;, 
                                 ytext =  &quot;Total proteins per cell&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Keep cells &gt;45 proteins&quot;,
                                 colorviolin = &quot;deeppink3&quot; ) +
                        geom_hline(yintercept = 45, size = 0.3) 

plot_percent.H3 &lt;- plot_QC_paper(seu_object = seu_combined, 
                                 feature = &quot;percent.HisH3&quot;, 
                                 ytext =   &quot;% Histone H3 counts&quot;,
                                 xtext = &quot;Time aIg stimulation (minutes)&quot;,
                                 paneltitle = &quot;Variation in % Histone H3 counts&quot;,
                                 colorviolin = &quot;deeppink3&quot; ) 
  
plot.QC &lt;- plot_grid(plot_RNA_nCount, plot_RNA_ngenes,  plot_percent.mt, plot_percent.rb,plot_PROT_nCount,plot_PROT_nproteins, plot_percent.H3, labels = c(&#39;a&#39;, &#39;b&#39;, &#39;c&#39;,&#39;d&#39; , &#39;e&#39;, &#39;f&#39;, &#39;g&#39;), label_size = 10, ncol = 2)

ggsave(plot.QC, filename = &quot;output/paper_figures/Suppl_QC_filters.pdf&quot;, width = 183, height = 200, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
ggsave(plot.QC, filename = &quot;output/paper_figures/Suppl_QC_filters.png&quot;, width = 183, height = 200, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>plot.QC</code></pre>
<p><img src="figure/QC.Rmd/Suppl_QC_filters-1.png" width="691.65312" style="display: block; margin: auto;" /> <em>Supplementary Figure</em> Thresholds for selection of high-quality samples and cells from the QuRIE-seq datasets.</p>
<p>Based on the indicated cut-offs, high-quality cells are filtered for further analysis.</p>
<pre class="r fold-show"><code>seu_combined_filtered &lt;- subset(seu_combined, subset = nFeature_RNA &gt; 150 &amp; nCount_RNA &lt; 4000 &amp; nFeature_PROT &gt; 45 &amp; nCount_PROT &lt; 3000 &amp; percent.mt &lt; 15)</code></pre>
<pre class="r"><code>## Small hack to prevent mofa error with duplicate gene or protein names
double &lt;- c(&quot;CD53&quot;, &quot;CD70&quot;, &quot;KLF6&quot;, &quot;XBP1&quot;)
PROT.counts &lt;-as.data.frame(seu_combined_filtered[[&quot;PROT&quot;]]@counts)
prot.rownames &lt;- rownames(PROT.counts)
row.names(PROT.counts) &lt;- ifelse(prot.rownames %in% double, paste0(prot.rownames, &quot;-PROT&quot;), paste0(prot.rownames))

seu_combined_filtered[[&quot;PROT&quot;]] &lt;- CreateAssayObject(counts = PROT.counts)</code></pre>
</div>
<div id="normalize-and-scale" class="section level2">
<h2>Normalize and scale</h2>
<pre class="r fold-show"><code># run sctransform with default settings.
seu_combined_filtered &lt;- SCTransform(seu_combined_filtered,
                            assay = &quot;RNA&quot;,
                            new.assay.name = &quot;SCT.RNA&quot;,
                            do.correct.umi = TRUE,
                            ncells = NULL,
                            variable.features.n = 3000,
                            vars.to.regress = c(&quot;percent.mt&quot;, &quot;nCount_RNA&quot;), # substantial variation between samples &amp; cells in mito and ncount
                            do.scale = FALSE,
                            do.center = TRUE,
                            conserve.memory = FALSE,
                            return.only.var.genes = FALSE,
                            seed.use = 42,
                            verbose = FALSE
                            )
# Add some metadata to normalized data (ncounts &amp; percent mt)
seu_combined_filtered &lt;- AddMetaData(seu_combined_filtered, as.data.frame(seu_combined_filtered@assays$SCT.RNA@counts) %&gt;% summarise_all(funs(sum)) %&gt;% unlist(), col.name = &quot;nCount_RNA_SCT&quot;)

seu_combined_filtered &lt;- PercentageFeatureSet(seu_combined_filtered, pattern = &quot;^MT\\.|^MTRN&quot;, col.name = &quot;percent.mt.aftersct&quot;, assay = &quot;SCT.RNA&quot;)

## cell cycle scoring metadata
s.genes &lt;- cc.genes$s.genes
g2m.genes &lt;- cc.genes$g2m.genes
seu_combined_filtered &lt;- CellCycleScoring(seu_combined_filtered, s.features = s.genes, g2m.features = g2m.genes, set.ident = FALSE, assay = &quot;SCT.RNA&quot;)

seu_combined_filtered[[&quot;S.score&quot;]] &lt;- seu_combined_filtered@meta.data$S.Score
seu_combined_filtered[[&quot;G2M.score&quot;]] &lt;- seu_combined_filtered@meta.data$G2M.Score
seu_combined_filtered[[&quot;CCphase&quot;]] &lt;- seu_combined_filtered@meta.data$Phase</code></pre>
<pre class="r"><code>all.prot &lt;- rownames(seu_combined_filtered[[&quot;PROT&quot;]])

seu_combined_filtered &lt;- NormalizeData(seu_combined_filtered, assay = &quot;PROT&quot;, normalization.method = &quot;CLR&quot;, verbose = FALSE)

seu_combined_filtered &lt;- ScaleData(seu_combined_filtered, assay = &quot;PROT&quot;, features = all.prot, vars.to.regress = c(&quot;nFeature_PROT&quot;,&quot;nCount_PROT&quot;, &quot;percent.HisH3&quot;))</code></pre>
<pre><code>Regressing out nFeature_PROT, nCount_PROT, percent.HisH3</code></pre>
<pre><code>Centering and scaling data matrix</code></pre>
<p>Seurat object with filtered cells and normalized counts is stored in “output/seu_combined_filtered_normalized.rds”</p>
<pre class="r fold-show"><code>saveRDS(seu_combined_filtered, &quot;output/seu_combined_filtered_normalized.rds&quot;)</code></pre>
</div>
<div id="subset-of-samples" class="section level2">
<h2>Subset of samples</h2>
<p>The manuscript describes two analysis of different collection of samples:<br />
* Effect of aIg stimulation over two time-scales (see MOFA aIg <a href="MOFAaIg.html">page</a>) * Effect of ibrutinib on the cell-state at these two timescales (see MOFA ibru <a href="MOFAibru.html">page</a>)</p>
<pre class="r fold-show"><code>seu_combined_aIg_selected &lt;- subset(seu_combined_filtered, idents = c(&quot;006.aIg.ibr&quot;, &quot;180.aIg.ibr&quot;), invert = TRUE)

saveRDS(seu_combined_aIg_selected, &quot;output/seu_aIG_samples.rds&quot;)</code></pre>
<pre class="r fold-show"><code>seu_combined_ibru_selected &lt;- subset(seu_combined_filtered, idents = c(&quot;002.aIg.contr&quot;,&quot;004.aIg.contr&quot;,&quot;060.aIg.contr&quot;), invert = TRUE)

saveRDS(seu_combined_ibru_selected, &quot;output/seu_ibru_samples.rds&quot;)</code></pre>
</div>
<div id="filtered-dataset-properties" class="section level2 tabset">
<h2 class="tabset">Filtered dataset properties</h2>
<p>Overview of the number of cells and data properties of all samples, aIg subset of samples, or ibrutinib subset of samples.</p>
<div id="full-dataset" class="section level3">
<h3>Full dataset</h3>
<pre class="r"><code>seu_combined_filtered</code></pre>
<pre><code>An object of class Seurat 
16824 features across 6952 samples within 3 assays 
Active assay: SCT.RNA (8372 features, 3000 variable features)
 2 other assays present: RNA, PROT</code></pre>
<p><strong>Table</strong> Overview of per sample properties after filtering</p>
<pre class="r"><code>kable(seu_combined_filtered@meta.data %&gt;% 
        group_by(condition) %&gt;% 
        summarise(`Number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  )) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
condition
</th>
<th style="text-align:right;">
Number of cells
</th>
<th style="text-align:right;">
Median counts RNA
</th>
<th style="text-align:right;">
Median Number genes
</th>
<th style="text-align:right;">
Median Mitochondrial counts (Median %)
</th>
<th style="text-align:right;">
Ribosomal counts (Median %)
</th>
<th style="text-align:right;">
Median counts PROT
</th>
<th style="text-align:right;">
Number proteins
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
000.aIg.contr
</td>
<td style="text-align:right;">
648
</td>
<td style="text-align:right;">
1393
</td>
<td style="text-align:right;">
964
</td>
<td style="text-align:right;">
7.18
</td>
<td style="text-align:right;">
6.78
</td>
<td style="text-align:right;">
1108
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
002.aIg.contr
</td>
<td style="text-align:right;">
923
</td>
<td style="text-align:right;">
783
</td>
<td style="text-align:right;">
599
</td>
<td style="text-align:right;">
7.69
</td>
<td style="text-align:right;">
6.86
</td>
<td style="text-align:right;">
993
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
004.aIg.contr
</td>
<td style="text-align:right;">
508
</td>
<td style="text-align:right;">
863
</td>
<td style="text-align:right;">
640
</td>
<td style="text-align:right;">
7.85
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
1042
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.contr
</td>
<td style="text-align:right;">
713
</td>
<td style="text-align:right;">
1087
</td>
<td style="text-align:right;">
794
</td>
<td style="text-align:right;">
8.13
</td>
<td style="text-align:right;">
7.03
</td>
<td style="text-align:right;">
749
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.ibr
</td>
<td style="text-align:right;">
943
</td>
<td style="text-align:right;">
493
</td>
<td style="text-align:right;">
386
</td>
<td style="text-align:right;">
10.46
</td>
<td style="text-align:right;">
6.05
</td>
<td style="text-align:right;">
777
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
060.aIg.contr
</td>
<td style="text-align:right;">
863
</td>
<td style="text-align:right;">
638
</td>
<td style="text-align:right;">
492
</td>
<td style="text-align:right;">
8.81
</td>
<td style="text-align:right;">
6.72
</td>
<td style="text-align:right;">
894
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.contr
</td>
<td style="text-align:right;">
1099
</td>
<td style="text-align:right;">
711
</td>
<td style="text-align:right;">
548
</td>
<td style="text-align:right;">
7.42
</td>
<td style="text-align:right;">
6.85
</td>
<td style="text-align:right;">
871
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.ibr
</td>
<td style="text-align:right;">
1255
</td>
<td style="text-align:right;">
732
</td>
<td style="text-align:right;">
571
</td>
<td style="text-align:right;">
6.37
</td>
<td style="text-align:right;">
6.96
</td>
<td style="text-align:right;">
810
</td>
<td style="text-align:right;">
57
</td>
</tr>
</tbody>
</table>
<p><strong>Table</strong> Overview of full filtered dataset properties.</p>
<pre class="r"><code>kable(seu_combined_filtered@meta.data %&gt;% 
        summarise(`Number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  ) %&gt;%
        t()) %&gt;%

  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
6952.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts RNA
</td>
<td style="text-align:right;">
741.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Number genes
</td>
<td style="text-align:right;">
566.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Mitochondrial counts (Median %)
</td>
<td style="text-align:right;">
7.90
</td>
</tr>
<tr>
<td style="text-align:left;">
Ribosomal counts (Median %)
</td>
<td style="text-align:right;">
6.71
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts PROT
</td>
<td style="text-align:right;">
876.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Number proteins
</td>
<td style="text-align:right;">
58.00
</td>
</tr>
</tbody>
</table>
</div>
<div id="aig-samples" class="section level3">
<h3>aIg samples</h3>
<pre class="r"><code>seu_combined_aIg_selected</code></pre>
<pre><code>An object of class Seurat 
16824 features across 4754 samples within 3 assays 
Active assay: SCT.RNA (8372 features, 3000 variable features)
 2 other assays present: RNA, PROT</code></pre>
<p><strong>Table</strong> Overview of aIg dataset properties per sample</p>
<pre class="r"><code>kable(seu_combined_aIg_selected@meta.data %&gt;% 
        group_by(condition) %&gt;% 
        summarise(`Number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  )) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
condition
</th>
<th style="text-align:right;">
Number of cells
</th>
<th style="text-align:right;">
Median counts RNA
</th>
<th style="text-align:right;">
Median Number genes
</th>
<th style="text-align:right;">
Median Mitochondrial counts (Median %)
</th>
<th style="text-align:right;">
Ribosomal counts (Median %)
</th>
<th style="text-align:right;">
Median counts PROT
</th>
<th style="text-align:right;">
Number proteins
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
000.aIg.contr
</td>
<td style="text-align:right;">
648
</td>
<td style="text-align:right;">
1393
</td>
<td style="text-align:right;">
964
</td>
<td style="text-align:right;">
7.18
</td>
<td style="text-align:right;">
6.78
</td>
<td style="text-align:right;">
1108
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
002.aIg.contr
</td>
<td style="text-align:right;">
923
</td>
<td style="text-align:right;">
783
</td>
<td style="text-align:right;">
599
</td>
<td style="text-align:right;">
7.69
</td>
<td style="text-align:right;">
6.86
</td>
<td style="text-align:right;">
993
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
004.aIg.contr
</td>
<td style="text-align:right;">
508
</td>
<td style="text-align:right;">
863
</td>
<td style="text-align:right;">
640
</td>
<td style="text-align:right;">
7.85
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
1042
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.contr
</td>
<td style="text-align:right;">
713
</td>
<td style="text-align:right;">
1087
</td>
<td style="text-align:right;">
794
</td>
<td style="text-align:right;">
8.13
</td>
<td style="text-align:right;">
7.03
</td>
<td style="text-align:right;">
749
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
060.aIg.contr
</td>
<td style="text-align:right;">
863
</td>
<td style="text-align:right;">
638
</td>
<td style="text-align:right;">
492
</td>
<td style="text-align:right;">
8.81
</td>
<td style="text-align:right;">
6.72
</td>
<td style="text-align:right;">
894
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.contr
</td>
<td style="text-align:right;">
1099
</td>
<td style="text-align:right;">
711
</td>
<td style="text-align:right;">
548
</td>
<td style="text-align:right;">
7.42
</td>
<td style="text-align:right;">
6.85
</td>
<td style="text-align:right;">
871
</td>
<td style="text-align:right;">
57
</td>
</tr>
</tbody>
</table>
<p><strong>Table</strong> Overview of aIg dataset properties.</p>
<pre class="r"><code>kable(seu_combined_aIg_selected@meta.data %&gt;% 
        summarise(`Number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  ) %&gt;%
        t()) %&gt;%

  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
4754.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts RNA
</td>
<td style="text-align:right;">
808.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Number genes
</td>
<td style="text-align:right;">
608.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Mitochondrial counts (Median %)
</td>
<td style="text-align:right;">
7.86
</td>
</tr>
<tr>
<td style="text-align:left;">
Ribosomal counts (Median %)
</td>
<td style="text-align:right;">
6.77
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts PROT
</td>
<td style="text-align:right;">
917.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Number proteins
</td>
<td style="text-align:right;">
58.00
</td>
</tr>
</tbody>
</table>
</div>
<div id="aig-ibrutinib-samples" class="section level3">
<h3>aIg + ibrutinib samples</h3>
<pre class="r"><code>seu_combined_ibru_selected</code></pre>
<pre><code>An object of class Seurat 
16824 features across 4658 samples within 3 assays 
Active assay: SCT.RNA (8372 features, 3000 variable features)
 2 other assays present: RNA, PROT</code></pre>
<p><strong>Table</strong> Overview of ibru dataset properties per sample</p>
<pre class="r"><code>kable(seu_combined_ibru_selected@meta.data %&gt;% 
        group_by(condition) %&gt;% 
        summarise(`Number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  )) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
condition
</th>
<th style="text-align:right;">
Number of cells
</th>
<th style="text-align:right;">
Median counts RNA
</th>
<th style="text-align:right;">
Median Number genes
</th>
<th style="text-align:right;">
Median Mitochondrial counts (Median %)
</th>
<th style="text-align:right;">
Ribosomal counts (Median %)
</th>
<th style="text-align:right;">
Median counts PROT
</th>
<th style="text-align:right;">
Number proteins
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
000.aIg.contr
</td>
<td style="text-align:right;">
648
</td>
<td style="text-align:right;">
1393
</td>
<td style="text-align:right;">
964
</td>
<td style="text-align:right;">
7.18
</td>
<td style="text-align:right;">
6.78
</td>
<td style="text-align:right;">
1108
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.contr
</td>
<td style="text-align:right;">
713
</td>
<td style="text-align:right;">
1087
</td>
<td style="text-align:right;">
794
</td>
<td style="text-align:right;">
8.13
</td>
<td style="text-align:right;">
7.03
</td>
<td style="text-align:right;">
749
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
006.aIg.ibr
</td>
<td style="text-align:right;">
943
</td>
<td style="text-align:right;">
493
</td>
<td style="text-align:right;">
386
</td>
<td style="text-align:right;">
10.46
</td>
<td style="text-align:right;">
6.05
</td>
<td style="text-align:right;">
777
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.contr
</td>
<td style="text-align:right;">
1099
</td>
<td style="text-align:right;">
711
</td>
<td style="text-align:right;">
548
</td>
<td style="text-align:right;">
7.42
</td>
<td style="text-align:right;">
6.85
</td>
<td style="text-align:right;">
871
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
180.aIg.ibr
</td>
<td style="text-align:right;">
1255
</td>
<td style="text-align:right;">
732
</td>
<td style="text-align:right;">
571
</td>
<td style="text-align:right;">
6.37
</td>
<td style="text-align:right;">
6.96
</td>
<td style="text-align:right;">
810
</td>
<td style="text-align:right;">
57
</td>
</tr>
</tbody>
</table>
<p><strong>Table</strong> Overview of ibru dataset properties.</p>
<pre class="r"><code>kable(seu_combined_ibru_selected@meta.data %&gt;% 
        summarise(`Number of cells` = round(n(),0),
                  `Median counts RNA` = round(median(nCount_RNA),0),
                  `Median Number genes` = round(median(nFeature_RNA),0),
                  `Median Mitochondrial counts (Median %)` = round(median(percent.mt),2), 
                  `Ribosomal counts (Median %)` = round(median(percent.rb),2),
                  `Median counts PROT` = round(median(nCount_PROT),0),
                  `Number proteins` = round(median(nFeature_PROT),0)
                  ) %&gt;%
        t()) %&gt;%

  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;))</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
4658.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts RNA
</td>
<td style="text-align:right;">
752.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Number genes
</td>
<td style="text-align:right;">
574.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Median Mitochondrial counts (Median %)
</td>
<td style="text-align:right;">
7.75
</td>
</tr>
<tr>
<td style="text-align:left;">
Ribosomal counts (Median %)
</td>
<td style="text-align:right;">
6.74
</td>
</tr>
<tr>
<td style="text-align:left;">
Median counts PROT
</td>
<td style="text-align:right;">
840.00
</td>
</tr>
<tr>
<td style="text-align:left;">
Number proteins
</td>
<td style="text-align:right;">
57.00
</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   
[3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.1252    

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] png_0.1-7                   forcats_0.5.1              
 [3] clusterProfiler_3.18.1      clusterProfiler.dplyr_0.0.2
 [5] enrichplot_1.10.2           org.Hs.eg.db_3.12.0        
 [7] AnnotationDbi_1.52.0        IRanges_2.24.1             
 [9] S4Vectors_0.28.1            Biobase_2.50.0             
[11] BiocGenerics_0.36.0         ggridges_0.5.3             
[13] cowplot_1.1.1               ggtext_0.1.1               
[15] ggplotify_0.0.5             ggcorrplot_0.1.3           
[17] ggrepel_0.9.1               ggpubr_0.4.0               
[19] scico_1.2.0                 MOFA2_1.1.17               
[21] extrafont_0.17              patchwork_1.1.1            
[23] RColorBrewer_1.1-2          viridis_0.5.1              
[25] viridisLite_0.3.0           ggsci_2.9                  
[27] sctransform_0.3.2           ggthemes_4.2.4             
[29] matrixStats_0.57.0          kableExtra_1.3.1           
[31] gridExtra_2.3               SeuratObject_4.0.0         
[33] Seurat_4.0.0                ggplot2_3.3.3              
[35] scales_1.1.1                tidyr_1.1.2                
[37] dplyr_1.0.3                 stringr_1.4.0              
[39] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.2       scattermore_0.7      bit64_4.0.5         
  [4] knitr_1.31           irlba_2.3.3          DelayedArray_0.16.1 
  [7] data.table_1.13.6    rpart_4.1-15         generics_0.1.0      
 [10] RSQLite_2.2.3        shadowtext_0.0.7     RANN_2.6.1          
 [13] future_1.21.0        bit_4.0.4            spatstat.data_2.1-0 
 [16] webshot_0.5.2        xml2_1.3.2           httpuv_1.5.5        
 [19] assertthat_0.2.1     xfun_0.23            hms_1.0.0           
 [22] evaluate_0.14        promises_1.1.1       readxl_1.3.1        
 [25] igraph_1.2.6         DBI_1.1.1            htmlwidgets_1.5.3   
 [28] purrr_0.3.4          ellipsis_0.3.1       corrplot_0.84       
 [31] backports_1.2.1      deldir_0.2-10        MatrixGenerics_1.2.0
 [34] vctrs_0.3.6          ROCR_1.0-11          abind_1.4-5         
 [37] cachem_1.0.1         withr_2.4.1          ggforce_0.3.2       
 [40] goftest_1.2-2        cluster_2.1.0        DOSE_3.16.0         
 [43] lazyeval_0.2.2       crayon_1.3.4         basilisk.utils_1.2.1
 [46] labeling_0.4.2       pkgconfig_2.0.3      tweenr_1.0.1        
 [49] nlme_3.1-149         rlang_0.4.10         globals_0.14.0      
 [52] lifecycle_0.2.0      miniUI_0.1.1.1       downloader_0.4      
 [55] filelock_1.0.2       extrafontdb_1.0      cellranger_1.1.0    
 [58] rprojroot_2.0.2      polyclip_1.10-0      lmtest_0.9-38       
 [61] Matrix_1.2-18        carData_3.0-4        Rhdf5lib_1.12.1     
 [64] zoo_1.8-8            whisker_0.4          pheatmap_1.0.12     
 [67] KernSmooth_2.23-17   rhdf5filters_1.2.0   blob_1.2.1          
 [70] qvalue_2.22.0        parallelly_1.23.0    rstatix_0.6.0       
 [73] gridGraphics_0.5-1   ggsignif_0.6.0       memoise_2.0.0       
 [76] magrittr_2.0.1       plyr_1.8.6           ica_1.0-2           
 [79] compiler_4.0.3       scatterpie_0.1.5     fitdistrplus_1.1-3  
 [82] listenv_0.8.0        pbapply_1.4-3        MASS_7.3-53         
 [85] mgcv_1.8-33          tidyselect_1.1.0     stringi_1.5.3       
 [88] highr_0.8            yaml_2.2.1           GOSemSim_2.16.1     
 [91] fastmatch_1.1-0      tools_4.0.3          future.apply_1.7.0  
 [94] rio_0.5.16           rstudioapi_0.13      foreign_0.8-80      
 [97] git2r_0.28.0         farver_2.0.3         Rtsne_0.15          
[100] ggraph_2.0.5         digest_0.6.27        rvcheck_0.1.8       
[103] BiocManager_1.30.10  shiny_1.6.0          Rcpp_1.0.6          
[106] gridtext_0.1.4       car_3.0-10           broom_0.7.3         
[109] later_1.1.0.1        RcppAnnoy_0.0.18     httr_1.4.2          
[112] colorspace_2.0-0     rvest_0.3.6          fs_1.5.0            
[115] tensor_1.5           reticulate_1.18      splines_4.0.3       
[118] uwot_0.1.10          spatstat.utils_2.1-0 graphlayouts_0.7.1  
[121] basilisk_1.2.1       plotly_4.9.3         xtable_1.8-4        
[124] jsonlite_1.7.2       spatstat_1.64-1      tidygraph_1.2.0     
[127] R6_2.5.0             pillar_1.4.7         htmltools_0.5.1.1   
[130] mime_0.9             glue_1.4.2           fastmap_1.1.0       
[133] BiocParallel_1.24.1  codetools_0.2-16     fgsea_1.16.0        
[136] lattice_0.20-41      tibble_3.0.5         curl_4.3            
[139] leiden_0.3.7         zip_2.1.1            GO.db_3.12.1        
[142] openxlsx_4.2.3       Rttf2pt1_1.3.8       survival_3.2-7      
[145] rmarkdown_2.6        munsell_0.5.0        DO.db_2.9           
[148] rhdf5_2.34.0         HDF5Array_1.18.0     haven_2.3.1         
[151] reshape2_1.4.4       gtable_0.3.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
